<!doctype html>
<html class="no-js" lang="en" data-content_root="../">
  <head><meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="index" title="Index" href="../genindex.html"><link rel="search" title="Search" href="../search.html">

    <!-- Generated with Sphinx 9.1.0 and Furo 2025.12.19 -->
        <title>art_modern: Modernized ART Simulator of Diverse Next-Generation Sequencing Reads - art_modern 1.3.4 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=d111a655" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo.css?v=7bdb33bb" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo-extensions.css?v=8dab3a3b" />
    
    


<style>
  body {
    --color-code-background: #f2f2f2;
  --color-code-foreground: #1e1e1e;
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-with-moon" viewBox="0 0 24 24">
    <title>Auto light/dark, in light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path style="opacity: 50%" d="M 5.411 14.504 C 5.471 14.504 5.532 14.504 5.591 14.504 C 3.639 16.319 4.383 19.569 6.931 20.352 C 7.693 20.586 8.512 20.551 9.25 20.252 C 8.023 23.207 4.056 23.725 2.11 21.184 C 0.166 18.642 1.702 14.949 4.874 14.536 C 5.051 14.512 5.231 14.5 5.411 14.5 L 5.411 14.504 Z"/>
      <line x1="14.5" y1="3.25" x2="14.5" y2="1.25"/>
      <line x1="14.5" y1="15.85" x2="14.5" y2="17.85"/>
      <line x1="10.044" y1="5.094" x2="8.63" y2="3.68"/>
      <line x1="19" y1="14.05" x2="20.414" y2="15.464"/>
      <line x1="8.2" y1="9.55" x2="6.2" y2="9.55"/>
      <line x1="20.8" y1="9.55" x2="22.8" y2="9.55"/>
      <line x1="10.044" y1="14.006" x2="8.63" y2="15.42"/>
      <line x1="19" y1="5.05" x2="20.414" y2="3.636"/>
      <circle cx="14.5" cy="9.55" r="3.6"/>
    </svg>
  </symbol>
  <symbol id="svg-moon-with-sun" viewBox="0 0 24 24">
    <title>Auto light/dark, in dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path d="M 8.282 7.007 C 8.385 7.007 8.494 7.007 8.595 7.007 C 5.18 10.184 6.481 15.869 10.942 17.24 C 12.275 17.648 13.706 17.589 15 17.066 C 12.851 22.236 5.91 23.143 2.505 18.696 C -0.897 14.249 1.791 7.786 7.342 7.063 C 7.652 7.021 7.965 7 8.282 7 L 8.282 7.007 Z"/>
      <line style="opacity: 50%" x1="18" y1="3.705" x2="18" y2="2.5"/>
      <line style="opacity: 50%" x1="18" y1="11.295" x2="18" y2="12.5"/>
      <line style="opacity: 50%" x1="15.316" y1="4.816" x2="14.464" y2="3.964"/>
      <line style="opacity: 50%" x1="20.711" y1="10.212" x2="21.563" y2="11.063"/>
      <line style="opacity: 50%" x1="14.205" y1="7.5" x2="13.001" y2="7.5"/>
      <line style="opacity: 50%" x1="21.795" y1="7.5" x2="23" y2="7.5"/>
      <line style="opacity: 50%" x1="15.316" y1="10.184" x2="14.464" y2="11.036"/>
      <line style="opacity: 50%" x1="20.711" y1="4.789" x2="21.563" y2="3.937"/>
      <circle style="opacity: 50%" cx="18" cy="7.5" r="2.169"/>
    </svg>
  </symbol>
  <symbol id="svg-pencil" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-pencil-code">
      <path d="M4 20h4l10.5 -10.5a2.828 2.828 0 1 0 -4 -4l-10.5 10.5v4" />
      <path d="M13.5 6.5l4 4" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
  <symbol id="svg-eye" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-eye-code">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M10 12a2 2 0 1 0 4 0a2 2 0 0 0 -4 0" />
      <path
        d="M11.11 17.958c-3.209 -.307 -5.91 -2.293 -8.11 -5.958c2.4 -4 5.4 -6 9 -6c3.6 0 6.6 2 9 6c-.21 .352 -.427 .688 -.647 1.008" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle site navigation sidebar">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc" aria-label="Toggle table of contents sidebar">
<label class="overlay sidebar-overlay" for="__navigation"></label>
<label class="overlay toc-overlay" for="__toc"></label>

<a class="skip-to-content muted-link" href="#furo-main-content">Skip to content</a>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <span class="icon"><svg><use href="#svg-menu"></use></svg></span>
      </label>
    </div>
    <div class="header-center">
      <a href="../index.html"><div class="brand">art_modern 1.3.4 documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle" aria-label="Toggle Light / Dark / Auto color theme">
          <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
          <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <span class="icon"><svg><use href="#svg-toc"></use></svg></span>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../index.html">
  
  <span class="sidebar-brand-text">art_modern 1.3.4 documentation</span>
  
</a><form class="sidebar-search-container" method="get" action="../search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="README_NEW.html"><code class="docutils literal notranslate"><span class="pre">art_modern</span></code>: Modernized ART Simulator of Diverse Next-Generation Sequencing Reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="docs/Install.html">Installing <code class="docutils literal notranslate"><span class="pre">art_modern</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="docs/Usage-art_modern.html">Usage of <code class="docutils literal notranslate"><span class="pre">art_modern</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="docs/Usage-apb.html">Usage of <code class="docutils literal notranslate"><span class="pre">art_profile_builder</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="docs/Usage-amc.html">Usage of <code class="docutils literal notranslate"><span class="pre">am_compress</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="docs/Guide.html">Guide on Miscellaneous Topics</a></li>
<li class="toctree-l1"><a class="reference internal" href="docs/FAQ.html">Frequently Asked Questions (FAQs)</a></li>
<li class="toctree-l1"><a class="reference internal" href="docs/Design.html">Design Topics</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="docs/Contributing.html">Developing &amp; Contributing</a><input aria-label="Toggle navigation of Developing &amp; Contributing" class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><span class="icon"><svg><use href="#svg-arrow-right"></use></svg></span></label><ul>
<li class="toctree-l2"><a class="reference internal" href="docs/MakefileTargets.html">Toplevel Makefile Documentation</a></li>
<li class="toctree-l2"><a class="reference internal" href="docs/TODO.html">TODO</a></li>
<li class="toctree-l2"><a class="reference internal" href="docs/CODE_OF_CONDUCT.html">Code of Conduct</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="docs/Security.html">Security Policy</a></li>
<li class="toctree-l1"><a class="reference internal" href="docs/News.html">News &amp; Release Notes</a></li>
<li class="toctree-l1"><a class="reference internal" href="docs/Copying.html">Copyright</a></li>
<li class="toctree-l1"><a class="reference internal" href="License.html">GNU GENERAL PUBLIC LICENSE</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          <div class="view-this-page">
  <a class="muted-link" href="../_sources/src/Readme.md.txt" title="View this page">
    <svg><use href="#svg-eye"></use></svg>
    <span class="visually-hidden">View this page</span>
  </a>
</div>
<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle" aria-label="Toggle Light / Dark / Auto color theme">
              <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
              <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <span class="icon"><svg><use href="#svg-toc"></use></svg></span>
          </label>
        </div>
        <article role="main" id="furo-main-content">
          <section class="tex2jax_ignore mathjax_ignore" id="art-modern-modernized-art-simulator-of-diverse-next-generation-sequencing-reads">
<h1><code class="docutils literal notranslate"><span class="pre">art_modern</span></code>: Modernized ART Simulator of Diverse Next-Generation Sequencing Reads<a class="headerlink" href="#art-modern-modernized-art-simulator-of-diverse-next-generation-sequencing-reads" title="Link to this heading">¶</a></h1>
<p><code class="docutils literal notranslate"><span class="pre">art_modern</span></code> badges:
<a class="reference external" href="https://github.com/YU-Zhejian/art_modern/releases"><img alt="GitHub Release" src="https://img.shields.io/github/v/release/YU-Zhejian/art_modern.svg" /></a>
<a class="reference external" href="https://github.com/YU-Zhejian/art_modern/releases/"><img alt="GitHub Downloads" src="https://img.shields.io/github/downloads/YU-Zhejian/art_modern/total.svg?style=social&amp;logo=github&amp;label=Download" /></a>
<a class="reference external" href="https://www.gnu.org/licenses/"><img alt="License" src="https://img.shields.io/badge/licence-GPL_3.0-blue.svg" /></a>
<a class="reference external" href="https://webkit.org/code-style-guidelines/"><img alt="Code style: WebKit" src="https://img.shields.io/badge/code%20style-webkit-blue.svg" /></a></p>
<p><code class="docutils literal notranslate"><span class="pre">art_modern</span></code> on BioConda: <a class="reference external" href="http://bioconda.github.io/recipes/art_modern/README.html"><img alt="Install with BioConda" src="https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg" /></a>
<a class="reference external" href="https://anaconda.org/bioconda/art_modern"><img alt="Conda Version" src="https://img.shields.io/conda/vn/bioconda/art_modern" /></a>
<a class="reference external" href="https://anaconda.org/bioconda/art_modern"><img alt="Conda Downloads" src="https://img.shields.io/conda/dn/bioconda/art_modern" /></a></p>
<p><code class="docutils literal notranslate"><span class="pre">art_modern-openmpi</span></code> on BioConda: <a class="reference external" href="http://bioconda.github.io/recipes/art_modern-openmpi/README.html"><img alt="Install with BioConda" src="https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg" /></a>
<a class="reference external" href="https://anaconda.org/bioconda/art_modern-openmpi"><img alt="Conda Version" src="https://img.shields.io/conda/vn/bioconda/art_modern-openmpi" /></a>
<a class="reference external" href="https://anaconda.org/bioconda/art_modern-openmpi"><img alt="Conda Downloads" src="https://img.shields.io/conda/dn/bioconda/art_modern-openmpi" /></a></p>
<p><code class="docutils literal notranslate"><span class="pre">art-modern-utils</span></code> badges:
<a class="reference external" href="https://github.com/YU-Zhejian/art_modern_utils/releases"><img alt="GitHub Release" src="https://img.shields.io/github/v/release/YU-Zhejian/art_modern_utils.svg" /></a>
<a class="reference external" href="https://www.gnu.org/licenses/"><img alt="License" src="https://img.shields.io/badge/licence-GPL_3.0-blue.svg" /></a>
<a class="reference external" href="https://github.com/psf/black"><img alt="Code style: black" src="https://img.shields.io/badge/code%20style-black-000000.svg" /></a></p>
<p><code class="docutils literal notranslate"><span class="pre">art-modern-utils</span></code> on PyPI:
<a class="reference external" href="https://www.python.org/downloads/"><img alt="Python version" src="https://img.shields.io/badge/python-3.9-blue" /></a>
<a class="reference external" href="https://pypi.org/project/art-modern-utils/"><img alt="PyPI - Version" src="https://img.shields.io/pypi/v/art-modern-utils" /></a>
<a class="reference external" href="https://pypi.org/project/art-modern-utils/"><img alt="PyPI - Downloads" src="https://img.shields.io/pypi/dm/art-modern-utils" /></a></p>
<p><a class="reference external" href="https://yu-zhejian.github.io/art_modern_docs/index.html"><img alt="Docs on GitHub Pages" src="https://img.shields.io/badge/docs-github--pages-blue?logo=github" /></a>
<a class="reference external" href="https://doi.org/10.64898/2026.02.20.707060"><img alt="bioRxiv" src="https://img.shields.io/static/v1?label=preprint&amp;message=bioRxiv&amp;color=ee4d2d" /></a></p>
<p><a class="reference external" href="https://surveillance.cancer.gov/genetic-simulation-resources/packages/art_modern/"><img alt="Catalogued on GSR" src="https://surveillance.cancer.gov/gsr/static/img/gsr_tile.jpg" /></a></p>
<section id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Link to this heading">¶</a></h2>
<p>Here we introduce <code class="docutils literal notranslate"><span class="pre">art_modern</span></code>, a modern re-implementation of the popular <a class="reference external" href="https://www.niehs.nih.gov/research/resources/software/biostatistics/art">ART</a> simulator with enhanced performance and functionality. It can be used for anyone who wants to simulate sequencing data for their own research, like benchmarking of DNA- or RNA-Seq alignment algorithms, test whether the RNA-Seq pipeline built by your lab performs well or perform pressure testing of pipelines on a cluster.</p>
<p><code class="docutils literal notranslate"><span class="pre">art_modern</span></code> supports ART-compatible error profiles. We also implemented a high-performance profile creator that creates ART-compatible profiles out of FASTQ, SAM, and BAM files.</p>
</section>
<section id="quick-start">
<h2>Quick Start<a class="headerlink" href="#quick-start" title="Link to this heading">¶</a></h2>
<section id="installation">
<h3>Installation<a class="headerlink" href="#installation" title="Link to this heading">¶</a></h3>
<p><strong>NOTE</strong> Although we’re providing compiled binaries for Linux, we still recommend compiling the source code yourself for maximum performance.</p>
<section id="using-fully-static-build">
<h4>Using Fully Static Build<a class="headerlink" href="#using-fully-static-build" title="Link to this heading">¶</a></h4>
<p>In each release, there will be a file named <code class="docutils literal notranslate"><span class="pre">build_rel_with_dbg_alpine-x86_64.tar.gz</span></code> in the <a class="reference external" href="https://github.com/YU-Zhejian/art_modern/releases">Releases</a> section. The file contains fully static linked executable binaries built under x86_64 Alpine Linux, which should work on most x86_64 Linux distributions. Unzip it, and you’re good to go.</p>
<p><strong>WARNING</strong> Static builds may lead to compromised security.</p>
<p><strong>NOTE</strong> The fully static build supports <strong>NEITHER</strong> MPI <strong>NOR</strong> NCBI SRA files.</p>
</section>
<section id="using-dpkg">
<h4>Using <code class="docutils literal notranslate"><span class="pre">dpkg</span></code><a class="headerlink" href="#using-dpkg" title="Link to this heading">¶</a></h4>
<p>If you use specific versions of Debian or Ubuntu, you can install <code class="docutils literal notranslate"><span class="pre">art_modern</span></code> through pre-built DEB packages available with each release. Root privileges are required.</p>
<p><strong>NOTE</strong> The DEB build does <strong>NOT</strong> support NCBI SRA files.</p>
</section>
<section id="using-docker-podman-singularity">
<h4>Using Docker/Podman/Singularity<a class="headerlink" href="#using-docker-podman-singularity" title="Link to this heading">¶</a></h4>
<p>The Docker/Podman/Singularity containers generated by BioConda is available at <a class="reference external" href="https://quay.io/repository/biocontainers/art_modern">Quay.io</a> (<a class="reference external" href="https://quay.io/repository/biocontainers/art_modern-openmpi">MPI-enabled version</a>).</p>
<p>A singularity example is provided below:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>singularity<span class="w"> </span>pull<span class="w"> </span>--name<span class="w"> </span>art_modern-1.3.2.sif<span class="w"> </span>docker://quay.io/biocontainers/art_modern:1.3.2--hb45bfb9_0
singularity<span class="w"> </span>run<span class="w"> </span>art_modern-1.3.2.sif<span class="w"> </span>art_modern<span class="w"> </span>--version
</pre></div>
</div>
</section>
<section id="using-conda">
<h4>Using Conda<a class="headerlink" href="#using-conda" title="Link to this heading">¶</a></h4>
<p><a class="reference external" href="https://docs.conda.io/">Conda</a> (or <a class="reference external" href="https://mamba.readthedocs.io/en/latest/">Mamba</a>/<a class="reference external" href="https://mamba.readthedocs.io/en/latest/user_guide/micromamba.html">micromamba</a>) is a popular open-source package and environment management system that simplifies the installation and management of software packages and their dependencies. Before processing, make sure you’ve installed Conda &gt;=25.7.0 by <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">--version</span></code>. Then, to create an environment named <code class="docutils literal notranslate"><span class="pre">art_modern_bioconda</span></code> with the package installed:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>create<span class="w"> </span>-y<span class="w"> </span>-n<span class="w"> </span>art_modern_bioconda<span class="w"> </span>-c<span class="w"> </span>bioconda<span class="w"> </span>-c<span class="w"> </span>conda-forge<span class="w"> </span>art_modern
conda<span class="w"> </span>run<span class="w"> </span>--no-capture-output<span class="w"> </span>--live-stream<span class="w"> </span>-n<span class="w"> </span>art_modern_bioconda<span class="w"> </span>art_modern<span class="w"> </span>--version
</pre></div>
</div>
<p>Or, use <code class="docutils literal notranslate"><span class="pre">art_modern-openmpi</span></code> to install the MPI-enabled version.</p>
<p><strong>NOTE</strong> The Conda build does <strong>NOT</strong> support NCBI SRA files.</p>
</section>
<section id="compiling-the-source-code">
<h4>Compiling the Source Code<a class="headerlink" href="#compiling-the-source-code" title="Link to this heading">¶</a></h4>
<p>Use <a class="reference external" href="https://git-scm.com/">Git</a> to clone this repository:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>git<span class="w"> </span>clone<span class="w"> </span>-b<span class="w"> </span>master<span class="w"> </span>https://github.com/YU-Zhejian/art_modern.git
<span class="nb">cd</span><span class="w"> </span>art_modern
</pre></div>
</div>
<p>Or, if you have no Git:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>curl<span class="w"> </span>-L<span class="w"> </span>-o<span class="w"> </span>art_modern-master.zip<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>https://github.com/YU-Zhejian/art_modern/archive/refs/heads/master.zip
<span class="c1"># -L used to make cURL following redirects</span>
unzip<span class="w"> </span>art_modern-master.zip
<span class="nb">cd</span><span class="w"> </span>art_modern-master
</pre></div>
</div>
<p><strong>STRONGLY DISCOURAGED</strong> Change <code class="docutils literal notranslate"><span class="pre">master</span></code> to <code class="docutils literal notranslate"><span class="pre">devel</span></code> to compile the development version</p>
<p>Ensure you have a C++ compiler that supports <a class="reference external" href="https://en.cppreference.com/w/cpp/17">C++17</a> installed on your computer (Preferably <a class="reference external" href="https://gcc.gnu.org/">GCC</a> &gt;= 9.5.0). Also check whether your <a class="reference external" href="https://cmake.org/">CMake</a>, <a class="reference external" href="https://www.gnu.org/software/make/">GNU Make</a>, <a class="reference external" href="https://www.boost.org/">Boost C++ Library</a>, <a class="reference external" href="https://www.gnu.org/software/binutils/">GNU BinUtils</a>, <a class="reference external" href="https://www.gnu.org/software/bash/">GNU Bash</a>, <a class="reference external" href="https://www.gnu.org/software/coreutils/">GNU CoreUtils</a>, <a class="reference external" href="https://www.python.org/">Python 3</a>, and minimal HTSLib dependencies (namely, <a class="reference external" href="https://www.zlib.net/">zlib</a> and <a class="reference external" href="https://www.man7.org/linux/man-pages/man7/pthreads.7.html">pthread</a>) are working.</p>
<p>Build the project using:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>opt/build_release
env<span class="w"> </span>-C<span class="w"> </span>opt/build_release<span class="w"> </span>cmake<span class="w"> </span>-DCMAKE_BUILD_TYPE<span class="o">=</span>RelWithDebInfo<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span><span class="s2">&quot;</span>
cmake<span class="w"> </span>--build<span class="w"> </span>opt/build_release<span class="w"> </span>-j<span class="s2">&quot;</span><span class="k">$(</span>nproc<span class="k">)</span><span class="s2">&quot;</span>
</pre></div>
</div>
<p>The project binary will be available at <code class="docutils literal notranslate"><span class="pre">opt/build_release/art_modern</span></code>. Now we can test whether the program runs through:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>opt/build_release/art_modern<span class="w"> </span>--help
opt/build_release/art_modern<span class="w"> </span>--version
</pre></div>
</div>
</section>
</section>
<section id="simulating-wgs-data-using-e-coli-genome">
<h3>Simulating WGS Data using <em>E. Coli</em> Genome<a class="headerlink" href="#simulating-wgs-data-using-e-coli-genome" title="Link to this heading">¶</a></h3>
<p>Download <em>E. coli</em> reference genome from NCBI. Here, we’ll use the K-12 strand MG1655 sub-strand as an example.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>opt/data/
<span class="nv">ACCESSION</span><span class="o">=</span><span class="s2">&quot;GCF_000005845.2_ASM584v2&quot;</span>
<span class="nv">BASEURL</span><span class="o">=</span><span class="s2">&quot;https://ftp.ncbi.nlm.nih.gov/genomes/all&quot;</span>
wget<span class="w"> </span>-O<span class="w"> </span>opt/data/GCF_000005845.2_ASM584v2_genomic.fna.gz<span class="w"> </span>-4<span class="w"> </span><span class="se">\</span>
<span class="w">    </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">BASEURL</span><span class="si">}</span><span class="s2">&quot;</span>/GCF/000/005/845/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">ACCESSION</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">ACCESSION</span><span class="si">}</span><span class="s2">&quot;</span>_genomic.fna.gz<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>
gunzip<span class="w"> </span>-n<span class="w"> </span>-k<span class="w"> </span>opt/data/GCF_000005845.2_ASM584v2_genomic.fna.gz
<span class="c1"># -n used to supress restoration of original timestamp.</span>
</pre></div>
</div>
<p>Now we can simulate WGS data using <em>E. coli</em> reference genome. Let’s start with single-end sequencing using HiSeq 2500 (which is the default <code class="docutils literal notranslate"><span class="pre">art_modern</span></code> profile and can be set explicitly through <code class="docutils literal notranslate"><span class="pre">--builtin_qual_file</span> <span class="pre">HiSeq2500_150bp</span></code>) with 250 bp read length (which is the longest read lengths supported by the profile and can be set explicitly through <code class="docutils literal notranslate"><span class="pre">--read_len</span> <span class="pre">250</span></code>), 10X coverage, and all available threads (can be set explicitly through <code class="docutils literal notranslate"><span class="pre">--parallel</span></code>).</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>opt/build_release/art_modern<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--mode<span class="w"> </span>wgs<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--lc<span class="w"> </span>se<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-file<span class="w"> </span>opt/data/GCF_000005845.2_ASM584v2_genomic.fna<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-fastq<span class="w"> </span>opt/build_release/e_coli_wgs_se.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-fcov<span class="w"> </span><span class="m">10</span>
</pre></div>
</div>
<p>The generated FASTQ file will be at <code class="docutils literal notranslate"><span class="pre">opt/build_release/e_coli_wgs_se.fastq</span></code>.</p>
<p>We may also simulate paired-end data with the following configuration:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>opt/build_release/art_modern<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--mode<span class="w"> </span>wgs<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--lc<span class="w"> </span>pe<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-file<span class="w"> </span>opt/data/GCF_000005845.2_ASM584v2_genomic.fna<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-fastq<span class="w"> </span>opt/build_release/e_coli_wgs_pe.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-fcov<span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--pe_frag_dist_mean<span class="w"> </span><span class="m">300</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--pe_frag_dist_std_dev<span class="w"> </span><span class="m">50</span>
</pre></div>
</div>
<p>Please note that we have additionally specified a quality file for read 2 with the mean and standard deviation of fragment lengths, which are required for paired-end sequencing. The generated fragment length should follow a normal (Gaussian) distribution with desired mean and standard deviation.</p>
<p><strong>NOTE</strong> The generated FASTQ file needed to be split into two FASTQ files and sorted for most pipelines. See <a class="reference internal" href="docs/FAQ.html#fastq-split-section"><span class="std std-ref">this section in the FAQ</span></a> for more details.</p>
</section>
<section id="simulating-rna-seq-data-using-c-elegans-transcriptome">
<h3>Simulating RNA-Seq Data using <em>C. Elegans</em> Transcriptome<a class="headerlink" href="#simulating-rna-seq-data-using-c-elegans-transcriptome" title="Link to this heading">¶</a></h3>
<p>Simulating transcriptome is a little more complicated since each cDNA molecule has different counts. Strand-specific library technologies also generate RNA-Seq data on one strand only. You’re recommended to use <a class="reference external" href="https://pypi.org/project/YASIM">YASIM</a> or other high-level simulators to generate expression for each cDNA molecule. You can also convert transcript quantification results generated by <a class="reference external" href="https://subread.sourceforge.net/featureCounts.html">featureCounts</a>, <a class="reference external" href="https://htseq.readthedocs.io/en/latest/">htseq-count</a>, <a class="reference external" href="https://salmon.readthedocs.io/en/latest/">Salmon</a>, <a class="reference external" href="https://pachterlab.github.io/kallisto/">Kallisto</a> or <a class="reference external" href="https://github.com/alexdobin/STAR">STAR</a> to the format supported by <code class="docutils literal notranslate"><span class="pre">art_modern</span></code>. The unified coverage model (i.e., like WGS) is also supported.</p>
<p>Please note that cDNAs with insufficient length will be ignored. We also do not support circular RNA simulation.</p>
<section id="unified-coverage">
<h4>Unified Coverage<a class="headerlink" href="#unified-coverage" title="Link to this heading">¶</a></h4>
<p>Following example samples 1000 long transcripts from reference <em>C. Elegans</em> transcriptome from <a class="reference external" href="https://genome.ucsc.edu/">UCSC Genome Browser</a> and performs a simulation using 10X unified coverage. You need to install <a class="reference external" href="https://github.com/lh3/seqtk">seqtk</a> and <a class="reference external" href="https://bioinf.shenwei.me/seqkit">seqkit</a> to run this example:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>curl<span class="w"> </span>https://hgdownload.soe.ucsc.edu/goldenPath/ce11/bigZips/mrna.fa.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>gzip<span class="w"> </span>-cdf<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>seqkit<span class="w"> </span>seq<span class="w"> </span>-m<span class="w"> </span><span class="m">200</span><span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>seqtk<span class="w"> </span>sample<span class="w"> </span>/dev/stdin<span class="w"> </span><span class="m">1000</span><span class="w"> </span>&gt;<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa

opt/build_release/art_modern<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--mode<span class="w"> </span>trans<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--lc<span class="w"> </span>se<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-file<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-fastq<span class="w"> </span>opt/build_release/c_elegans_trans_unified_se.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-fcov<span class="w"> </span><span class="m">10</span>
</pre></div>
</div>
</section>
<section id="unstranded-coverage">
<h4>Unstranded Coverage<a class="headerlink" href="#unstranded-coverage" title="Link to this heading">¶</a></h4>
<p>To simulate data with unstranded coverage information (i.e., same coverage on both strands), you need to provide an additional TSV file with one column of transcript ID and another column of coverage (in floating points). Please note that lines started by <code class="docutils literal notranslate"><span class="pre">#</span></code> will be ignored. An example of the coverage file (Depth rounded to 2 decimal for simplicity):</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>NM_069135	6.69
NR_056112	5.19
NR_051843	3.45
NR_066512	4.73
</pre></div>
</div>
<p>The following example generates a coverage file using <a class="reference external" href="https://www.gnu.org/software/gawk/">GNU AWK</a> with random coverage ranging between 0 and 10 assigned to each cDNA molecule.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>samtools<span class="w"> </span>faidx<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa
awk<span class="w"> </span><span class="s1">&#39;BEGIN{print &quot;#ID\tCOV&quot;;}{printf &quot;%s\t%f\n&quot;, $1, (rand()*10);}&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>&lt;<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa.fai<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>&gt;<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa.unstranded_cov.tsv

opt/build_release/art_modern<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--mode<span class="w"> </span>trans<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--lc<span class="w"> </span>se<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-file<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-fastq<span class="w"> </span>opt/build_release/c_elegans_trans_unstranded_se.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-fcov<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa.unstranded_cov.tsv
</pre></div>
</div>
</section>
<section id="stranded-coverage">
<h4>Stranded Coverage<a class="headerlink" href="#stranded-coverage" title="Link to this heading">¶</a></h4>
<p>To simulate data with stranded coverage information (i.e., coverage on one strand is different from the other), you need to provide an additional TSV file with one column of transcript ID and two other columns of coverage in positive and negative strand (in floating points). An example of the coverage file:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>NM_069135	2.31	4.38
NR_056112	3.47	1.72
NR_051843	1.35	2.09
NR_066512	3.04	1.68
</pre></div>
</div>
<p>Code example:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{print &quot;#ID\tCOV_POS\tCOV_NEG&quot;;}{printf &quot;%s\t%f\t%f\n&quot;, $1, (rand()*5), (rand()*5);}&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>&lt;<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa.fai<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>&gt;<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa.stranded_cov.tsv

opt/build_release/art_modern<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--mode<span class="w"> </span>trans<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--lc<span class="w"> </span>se<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-file<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-fastq<span class="w"> </span>opt/build_release/c_elegans_trans_stranded_se.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-fcov<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa.stranded_cov.tsv
</pre></div>
</div>
</section>
<section id="the-pbsim3-transcripts-input-format">
<h4>The PBSIM3 Transcripts Input Format<a class="headerlink" href="#the-pbsim3-transcripts-input-format" title="Link to this heading">¶</a></h4>
<p>The PBSIM3 Transcripts input format is a 4-column tab-delimited text file with transcript ID, sequence, and coverage on both strands. This file includes both sequence and coverage, so no additional coverage parameter is required. Similarly, sequences with insufficient length and lines started with <code class="docutils literal notranslate"><span class="pre">#</span></code> will be ignored. An example of the transcript input file is as follows (Sequences represented as <code class="docutils literal notranslate"><span class="pre">aaaa</span></code>):</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>NR_056112	3.47	1.72	aaaa
NR_051843	1.35	2.09	aaaa
NR_066512	3.04	1.68	aaaa
NM_061905	0.96	1.39	aaaa
NR_054174	3.59	4.92	aaaa
</pre></div>
</div>
<p>The following example converts the FASTA file to the PBSIM3 Transcripts input format with the help of <a class="reference external" href="https://bioinf.shenwei.me/seqkit">seqkit</a> with random coverage generated using GNU AWK. Note that the <code class="docutils literal notranslate"><span class="pre">--i-type</span></code> parameter is required.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>seqkit<span class="w"> </span>fx2tab<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{print &quot;#ID\tCOV_POS\tCOV_NEG\tSEQ&quot;;}{printf &quot;%s\t%f\t%f\t%s\n&quot;, $1, (rand()*5), (rand()*5), $3;}&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>&gt;<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa.pbsim3_trans.tsv

opt/build_release/art_modern<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--mode<span class="w"> </span>trans<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--lc<span class="w"> </span>se<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-file<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa.pbsim3_trans.tsv<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-fastq<span class="w"> </span>opt/build_release/c_elegans_trans_pbsim3_se.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-type<span class="w"> </span>pbsim3_transcripts
</pre></div>
</div>
</section>
</section>
<section id="template-based-simulation">
<h3>Template-Based Simulation<a class="headerlink" href="#template-based-simulation" title="Link to this heading">¶</a></h3>
<p>Template-based simulation is often used to introduce Illumina specific errors to cDNA molecules generated from some upstream simulator like <a class="reference external" href="https://camparee.readthedocs.io/en/latest/">CAMPAREE</a>. In this mode, single-end reads will be started from the first base of the template while paired-end/mate-pair reads will span the entire template. The template-based simulation mode also supports the PBSIM3 Transcripts format. For example:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>opt/build_release/art_modern<span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--mode<span class="w"> </span>template<span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--lc<span class="w"> </span>pe<span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--i-file<span class="w"> </span>opt/build_release/ce11_mrna_1000.fa.pbsim3_trans.tsv<span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--o-fastq<span class="w"> </span>opt/build_release/c_elegans_template_pbsim3_se.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--i-type<span class="w"> </span>pbsim3_transcripts
</pre></div>
</div>
<p>Please note that the mean and standard deviation of fragment length is not specified since in template-based simulation, a template is considered a fragment.</p>
<p><strong>NOTE</strong> For coverage of the template mode:</p>
<ul class="simple">
<li><p>If a unified coverage file is provided, the coverage will be interpreted as positive coverage instead of being divided equally into both strands.</p></li>
<li><p>If a 2-column (unstranded) coverage file is provided, the coverage will be interpreted as positive coverage.</p></li>
<li><p>If a 3-column (stranded) coverage file or input in format of <code class="docutils literal notranslate"><span class="pre">pvsim3_transcripts</span></code> is provided, the coverage will be interpreted as-is.</p></li>
</ul>
</section>
</section>
<section id="advanced-topics">
<h2>Advanced Topics<a class="headerlink" href="#advanced-topics" title="Link to this heading">¶</a></h2>
<section id="using-unix-pipelines">
<h3>Using UNIX Pipelines<a class="headerlink" href="#using-unix-pipelines" title="Link to this heading">¶</a></h3>
<p>With UNIX pipelines, we can redirect the input and output of <code class="docutils literal notranslate"><span class="pre">art_modern</span></code> from/to another file/processes. Following example reads FASTA reference from <code class="docutils literal notranslate"><span class="pre">/dev/stdin</span></code> (Standard Input), and writes compressed FASTQ, FASTA, and sorted BAM files.</p>
<p>This example requires <a class="reference external" href="https://www.gnu.org/software/gzip/">gzip</a>, <a class="reference external" href="https://zlib.net/pigz/">pigz</a>, <a class="reference external" href="https://github.com/samtools/samtools">SAMtools</a>, and <a class="reference external" href="https://tukaani.org/xz/">XZ Utils</a>. Note that the <code class="docutils literal notranslate"><span class="pre">--i-type</span></code> parameter is required since now, the simulator cannot tell the file format through its file extension.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>zcat<span class="w"> </span>opt/data/GCF_000005845.2_ASM584v2_genomic.fna.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>opt/build_release/art_modern<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--mode<span class="w"> </span>wgs<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--lc<span class="w"> </span>se<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-file<span class="w"> </span>/dev/stdin<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-type<span class="w"> </span>fasta<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-parser<span class="w"> </span>memory<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-fastq<span class="w"> </span>&gt;<span class="o">(</span>pigz<span class="w"> </span>-9-cf<span class="w"> </span>-<span class="w"> </span>&gt;<span class="w"> </span>opt/build_release/e_coli_wgs_se.fastq.gz<span class="o">)</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-fasta<span class="w"> </span>&gt;<span class="o">(</span>xz<span class="w"> </span>-9<span class="w"> </span>-T5<span class="w"> </span>-cf<span class="w"> </span>-<span class="w"> </span>&gt;<span class="w"> </span>opt/build_release/e_coli_wgs_se.fasta.xz<span class="o">)</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-sam<span class="w"> </span>&gt;<span class="o">(</span>samtools<span class="w"> </span>sort<span class="w"> </span>-o<span class="w"> </span>opt/build_release/e_coli_wgs_se.sorted.bam<span class="o">)</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-fcov<span class="w"> </span><span class="m">5</span>
</pre></div>
</div>
<p>Please wait for a while for the compression to finish.</p>
<p><strong>NOTE</strong> Pipelines and redirections are not supported in MPI mode mentioned below.</p>
<p><strong>TIPS</strong> You may find program <a class="reference external" href="https://www.man7.org/linux/man-pages/man1/pv.1.html"><code class="docutils literal notranslate"><span class="pre">pv(1)</span></code></a> useful when working with pipelines.</p>
</section>
<section id="using-mpi">
<h3>Using MPI<a class="headerlink" href="#using-mpi" title="Link to this heading">¶</a></h3>
<p>Message Passing Interface (<a class="reference external" href="https://www.mpi-forum.org/">MPI</a>) is a standard for parallel computing on a high-performance cluster (HPC). It allows processes to communicate with each other through intranet. If you’re working on a cluster with MPI installed, you can use the MPI version of <code class="docutils literal notranslate"><span class="pre">art_modern</span></code> to speed up the simulation by executing <code class="docutils literal notranslate"><span class="pre">art_modern</span></code> on different nodes in parallel.</p>
<p><strong>NOTE</strong> Please consult your cluster administrator/cluster documentation before using MPI version, since some clusters may have special configurations and/or quota limitations.</p>
<p>See also:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://slurm.schedmd.com/mpi_guide.html">Documentation for MPI Integration in Slurm</a> if your cluster uses <a class="reference external" href="https://slurm.schedmd.com/">Slurm Workload Manager</a>.</p></li>
</ul>
<p>Documentations from MPI vendors:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://www.open-mpi.org/doc/">Open MPI Documentation</a>, especially <a class="reference external" href="https://docs.open-mpi.org/en/v5.0.x/launching-apps/index.html">Launching MPI applications</a> section.</p></li>
<li><p><a class="reference external" href="https://www.mpich.org/documentation/">MPICH Documentation</a>.</p></li>
<li><p><a class="reference external" href="https://www.intel.com/content/www/us/en/developer/tools/oneapi/mpi-library.html?wapkw=MPI">Intel MPI Documentation</a>.</p></li>
</ul>
<p>You may also find the following information helpful:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://openhpc.community/">OpenHPC Community</a> if you’d like to build your own HPC cluster.</p></li>
</ul>
<section id="building-mpi-enabled-art-modern">
<h4>Building MPI-Enabled <code class="docutils literal notranslate"><span class="pre">art_modern</span></code><a class="headerlink" href="#building-mpi-enabled-art-modern" title="Link to this heading">¶</a></h4>
<p>Load the MPI environment on your cluster. Then execute:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>opt/build_release-mpi
env<span class="w"> </span>-C<span class="w"> </span>opt/build_release-mpi<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>cmake<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>-DCMAKE_BUILD_TYPE<span class="o">=</span>RelWithDebInfo<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>-DWITH_MPI<span class="o">=</span>ON<span class="w"> </span><span class="se">\</span>
<span class="w">    </span><span class="s2">&quot;</span><span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span><span class="s2">&quot;</span>
cmake<span class="w"> </span>--build<span class="w"> </span>opt/build_release-mpi<span class="w"> </span>-j<span class="s2">&quot;</span><span class="k">$(</span>nproc<span class="k">)</span><span class="s2">&quot;</span>
</pre></div>
</div>
<p>Test whether the MPI version works:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mpiexec<span class="w"> </span>-n<span class="w"> </span><span class="m">1</span><span class="w"> </span>opt/build_release-mpi/art_modern-mpi<span class="w"> </span>--version
</pre></div>
</div>
<p>You should be able to see the output MPI standard version and vendor-specific information. For example:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>MPI:
        Standard Version: 3.1
        Library Version: Open MPI v4.1.6, [...]
</pre></div>
</div>
<p>If you’re using OpenMPI.</p>
</section>
<section id="changes-in-output-files">
<h4>Changes in Output Files<a class="headerlink" href="#changes-in-output-files" title="Link to this heading">¶</a></h4>
<p>Let us run a WGS simulation with 4 MPI processes:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mpiexec<span class="w"> </span>-n<span class="w"> </span><span class="m">4</span><span class="w"> </span>opt/build_release-mpi/art_modern-mpi<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--mode<span class="w"> </span>wgs<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--lc<span class="w"> </span>se<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-file<span class="w"> </span>opt/data/GCF_000005845.2_ASM584v2_genomic.fna<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-fastq<span class="w"> </span>opt/build_release-mpi/e_coli_wgs_se.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-fcov<span class="w"> </span><span class="m">20</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--parallel<span class="w"> </span><span class="m">4</span>
</pre></div>
</div>
<p>4 files, namely <code class="docutils literal notranslate"><span class="pre">e_coli_wgs_se.0.fastq</span></code>, <code class="docutils literal notranslate"><span class="pre">e_coli_wgs_se.1.fastq</span></code>, <code class="docutils literal notranslate"><span class="pre">e_coli_wgs_se.2.fastq</span></code>, and <code class="docutils literal notranslate"><span class="pre">e_coli_wgs_se.3.fastq</span></code> will be generated in <code class="docutils literal notranslate"><span class="pre">opt/build_release-mpi/</span></code> directory. Each file corresponds to the output from each MPI process (rank 0 to rank 3). This behavior is the same for other output files (e.g., PWA, SAM/BAM).</p>
<p><strong>NOTE</strong> In the above example, the actual number of computational threads used in simulation will be 16 (4 MPI processes * 4 threads per process) since we’ve specified <code class="docutils literal notranslate"><span class="pre">--parallel</span> <span class="pre">4</span></code>.</p>
</section>
<section id="other-differences-between-mpi-and-non-mpi-versions">
<h4>Other Differences Between MPI and Non-MPI Versions<a class="headerlink" href="#other-differences-between-mpi-and-non-mpi-versions" title="Link to this heading">¶</a></h4>
<ul class="simple">
<li><p>The simulator will not be able to support UNIX devices and/or redirections for input/output files. So, even if your input is enormous, you have to write it to a physical file first. However, as most HPC clusters use distributed file systems, this should not be a big problem.</p></li>
<li><p>Logging issues: Only log messages from rank 0 process will be printed to standard error. Other ranks’ log messages will be written to disk (If environment variable <code class="docutils literal notranslate"><span class="pre">ART_NO_LOG_DIR</span></code> is not set) or discarded (If environment variable <code class="docutils literal notranslate"><span class="pre">ART_NO_LOG_DIR</span></code> is set).</p></li>
</ul>
</section>
</section>
</section>
<section id="building-art-art-modern-profiles">
<h2>Building ART/<code class="docutils literal notranslate"><span class="pre">art_modern</span></code> Profiles<a class="headerlink" href="#building-art-art-modern-profiles" title="Link to this heading">¶</a></h2>
<p>The <code class="docutils literal notranslate"><span class="pre">art_profile_builder</span></code> is a new executable designed to replace the old art_profiler_illumina Shell/Perl scripts for building ART-compatible quality profiles. It supports input from FASTQ, SAM/BAM, and NCBI SRA (If configured) files. The generated files would be compatible with the original ART and <code class="docutils literal notranslate"><span class="pre">art_modern</span></code>.</p>
<p>Examples using single-end FASTQ:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>art_profile_builder<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-file<span class="w"> </span>input.fq<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--read_len<span class="w"> </span><span class="m">36</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-file1<span class="w"> </span>output_profile.txt<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--parallel<span class="w"> </span><span class="m">8</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-num_threads<span class="w"> </span><span class="m">4</span>
</pre></div>
</div>
<p>Examples using paired-end SAM/BAM:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>art_profile_builder<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-file<span class="w"> </span>input.sam<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--read_len<span class="w"> </span><span class="m">36</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--is_pe<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-file1<span class="w"> </span>output_R1.txt<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-file2<span class="w"> </span>output_R2.txt<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--parallel<span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-num_threads<span class="w"> </span><span class="m">4</span>
</pre></div>
</div>
<p>For complete usage information, see the <a class="reference internal" href="docs/Usage-apb.html#art-profile-builder-usage-section"><span class="std std-ref">full documentation</span></a>.</p>
<p>By default, the program would <strong>NOT</strong> support building profiles from NCBI short-read archive (SRA). Use <code class="docutils literal notranslate"><span class="pre">-DWITH_NCBI_NGS=ON</span></code> to enable it. Additional libraries are required. See <a class="reference internal" href="docs/Install.html#optional-ncbi-ngs-components"><span class="std std-ref">this section in Installation instruction</span></a> for details.</p>
</section>
<section id="what-s-next">
<h2>What’s Next?<a class="headerlink" href="#what-s-next" title="Link to this heading">¶</a></h2>
<p>The project provides diverse documentations to satisfy your needs.</p>
<ul class="simple">
<li><p>If you want to build the software with different options, see <a class="reference internal" href="docs/Install.html"><span class="std std-doc">Install</span></a>.</p></li>
<li><p>For a detailed guide on parameters and their combinations, see <a class="reference internal" href="docs/Usage-art_modern.html"><span class="std std-doc">Usage for <code class="docutils literal notranslate"><span class="pre">art_modern</span></code></span></a>, <a class="reference internal" href="docs/Usage-apb.html"><span class="std std-doc">Usage for <code class="docutils literal notranslate"><span class="pre">art_profile_builder</span></code></span></a> and <a class="reference internal" href="docs/FAQ.html"><span class="std std-doc">FAQ</span></a>.</p></li>
<li><p>For developers, please refer to:</p>
<ul>
<li><p><a class="reference internal" href="docs/Contributing.html"><span class="std std-doc">Contributing</span></a> for software engineering tasks and contribution guidelines. See also <a class="reference internal" href="docs/CODE_OF_CONDUCT.html"><span class="std std-doc">Code of Conduct</span></a>.</p></li>
<li><p><a class="reference internal" href="docs/Design.html"><span class="std std-doc">Design</span></a> for the latest design of the software.</p></li>
<li><p><a class="reference internal" href="docs/Copying.html"><span class="std std-doc">Copying</span></a> for third-party libraries and codes used in this project.</p></li>
<li><p><a class="reference internal" href="docs/News.html"><span class="std std-doc">News</span></a> for changes over the project.</p></li>
</ul>
</li>
<li><p>For a comparison of this project with other simulators, see <a class="reference external" href="https://github.com/YU-Zhejian/art_modern_benchmark_other_simulators">Benchmark</a>.</p></li>
<li><p>The original ART documentation is copied to <a class="reference external" href="https://github.com/YU-Zhejian/art_modern/tree/master/docs/original_art_docs">here</a>.</p></li>
</ul>
</section>
<section id="acknowledgements">
<h2>Acknowledgements<a class="headerlink" href="#acknowledgements" title="Link to this heading">¶</a></h2>
<p>This simulator is based on the works of <a class="reference external" href="mailto:whduke&#37;&#52;&#48;gmail&#46;com">Weichun Huang</a> <em>et al.</em>, under <a class="reference external" href="https://www.gnu.org/licenses/">GNU GPL v3</a> license. The software is originally distributed <a class="reference external" href="https://www.niehs.nih.gov/research/resources/software/biostatistics/art">here</a> with the following reference:</p>
<ul class="simple">
<li><p>W. Huang, L. Li, J. R. Myers, and G. T. Marth, <em>ART: a next-generation sequencing read simulator</em>, Bioinformatics (Oxford, England), vol. 28, no. 4, pp. 593–594, Feb. 2012, DOI: <a class="reference external" href="https://doi.org/10.1093/bioinformatics/btr708">10.1093/bioinformatics/btr708</a>.</p></li>
</ul>
<p>Bundled libraries in this project are distributed under their own licenses. See <a class="reference internal" href="docs/Copying.html"><span class="std std-doc">Copying</span></a> for details.</p>
</section>
<section id="citing-this-work">
<h2>Citing This Work<a class="headerlink" href="#citing-this-work" title="Link to this heading">¶</a></h2>
<p>Following is the bioRxiv preprint of this work.</p>
<div class="highlight-bibtex notranslate"><div class="highlight"><pre><span></span><span class="nc">@article</span><span class="w"> </span><span class="p">{</span><span class="nl">YU2026.02.20.707060</span><span class="p">,</span>
<span class="w">	</span><span class="na">author</span><span class="w"> </span><span class="p">=</span><span class="w"> </span><span class="s">{YU, Zhejian}</span><span class="p">,</span>
<span class="w">	</span><span class="na">title</span><span class="w"> </span><span class="p">=</span><span class="w"> </span><span class="s">{art_modern: An Accelerated ART Simulator of Diverse Next-Generation Sequencing Reads}</span><span class="p">,</span>
<span class="w">	</span><span class="na">elocation-id</span><span class="w"> </span><span class="p">=</span><span class="w"> </span><span class="s">{2026.02.20.707060}</span><span class="p">,</span>
<span class="w">	</span><span class="na">year</span><span class="w"> </span><span class="p">=</span><span class="w"> </span><span class="s">{2026}</span><span class="p">,</span>
<span class="w">	</span><span class="na">doi</span><span class="w"> </span><span class="p">=</span><span class="w"> </span><span class="s">{10.64898/2026.02.20.707060}</span><span class="p">,</span>
<span class="w">	</span><span class="na">publisher</span><span class="w"> </span><span class="p">=</span><span class="w"> </span><span class="s">{Cold Spring Harbor Laboratory}</span><span class="p">,</span>
<span class="w">	</span><span class="na">URL</span><span class="w"> </span><span class="p">=</span><span class="w"> </span><span class="s">{https://www.biorxiv.org/content/early/2026/02/23/2026.02.20.707060}</span><span class="p">,</span>
<span class="w">	</span><span class="na">eprint</span><span class="w"> </span><span class="p">=</span><span class="w"> </span><span class="s">{https://www.biorxiv.org/content/early/2026/02/23/2026.02.20.707060.full.pdf}</span><span class="p">,</span>
<span class="w">	</span><span class="na">journal</span><span class="w"> </span><span class="p">=</span><span class="w"> </span><span class="s">{bioRxiv}</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="stargazers-over-time">
<h2>Stargazers Over Time<a class="headerlink" href="#stargazers-over-time" title="Link to this heading">¶</a></h2>
<p><a class="reference external" href="https://starchart.cc/YU-Zhejian/art_modern"><img alt="Stargazers over time" src="https://starchart.cc/YU-Zhejian/art_modern.svg?variant=adaptive" /></a></p>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          
          
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2024-2026, YU Zhejian
            </div>
            Made with <a href="https://www.sphinx-doc.org/">Sphinx</a> and <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
            
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#"><code class="docutils literal notranslate"><span class="pre">art_modern</span></code>: Modernized ART Simulator of Diverse Next-Generation Sequencing Reads</a><ul>
<li><a class="reference internal" href="#introduction">Introduction</a></li>
<li><a class="reference internal" href="#quick-start">Quick Start</a><ul>
<li><a class="reference internal" href="#installation">Installation</a><ul>
<li><a class="reference internal" href="#using-fully-static-build">Using Fully Static Build</a></li>
<li><a class="reference internal" href="#using-dpkg">Using <code class="docutils literal notranslate"><span class="pre">dpkg</span></code></a></li>
<li><a class="reference internal" href="#using-docker-podman-singularity">Using Docker/Podman/Singularity</a></li>
<li><a class="reference internal" href="#using-conda">Using Conda</a></li>
<li><a class="reference internal" href="#compiling-the-source-code">Compiling the Source Code</a></li>
</ul>
</li>
<li><a class="reference internal" href="#simulating-wgs-data-using-e-coli-genome">Simulating WGS Data using <em>E. Coli</em> Genome</a></li>
<li><a class="reference internal" href="#simulating-rna-seq-data-using-c-elegans-transcriptome">Simulating RNA-Seq Data using <em>C. Elegans</em> Transcriptome</a><ul>
<li><a class="reference internal" href="#unified-coverage">Unified Coverage</a></li>
<li><a class="reference internal" href="#unstranded-coverage">Unstranded Coverage</a></li>
<li><a class="reference internal" href="#stranded-coverage">Stranded Coverage</a></li>
<li><a class="reference internal" href="#the-pbsim3-transcripts-input-format">The PBSIM3 Transcripts Input Format</a></li>
</ul>
</li>
<li><a class="reference internal" href="#template-based-simulation">Template-Based Simulation</a></li>
</ul>
</li>
<li><a class="reference internal" href="#advanced-topics">Advanced Topics</a><ul>
<li><a class="reference internal" href="#using-unix-pipelines">Using UNIX Pipelines</a></li>
<li><a class="reference internal" href="#using-mpi">Using MPI</a><ul>
<li><a class="reference internal" href="#building-mpi-enabled-art-modern">Building MPI-Enabled <code class="docutils literal notranslate"><span class="pre">art_modern</span></code></a></li>
<li><a class="reference internal" href="#changes-in-output-files">Changes in Output Files</a></li>
<li><a class="reference internal" href="#other-differences-between-mpi-and-non-mpi-versions">Other Differences Between MPI and Non-MPI Versions</a></li>
</ul>
</li>
</ul>
</li>
<li><a class="reference internal" href="#building-art-art-modern-profiles">Building ART/<code class="docutils literal notranslate"><span class="pre">art_modern</span></code> Profiles</a></li>
<li><a class="reference internal" href="#what-s-next">What’s Next?</a></li>
<li><a class="reference internal" href="#acknowledgements">Acknowledgements</a></li>
<li><a class="reference internal" href="#citing-this-work">Citing This Work</a></li>
<li><a class="reference internal" href="#stargazers-over-time">Stargazers Over Time</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="../_static/documentation_options.js?v=e358f374"></script>
    <script src="../_static/doctools.js?v=fd6eb6e6"></script>
    <script src="../_static/sphinx_highlight.js?v=6ffebe34"></script>
    <script src="../_static/scripts/furo.js?v=46bd48cc"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    </body>
</html>