<!doctype html>
<html class="no-js" lang="en" data-content_root="../../">
  <head><meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="index" title="Index" href="../../genindex.html"><link rel="search" title="Search" href="../../search.html"><link rel="next" title="Usage of art_profile_builder" href="Usage-apb.html"><link rel="prev" title="Installing art_modern" href="Install.html">

    <!-- Generated with Sphinx 9.1.0 and Furo 2025.12.19 -->
        <title>Usage of art_modern - art_modern 1.3.4 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=d111a655" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/furo.css?v=7bdb33bb" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/furo-extensions.css?v=8dab3a3b" />
    
    


<style>
  body {
    --color-code-background: #f2f2f2;
  --color-code-foreground: #1e1e1e;
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-with-moon" viewBox="0 0 24 24">
    <title>Auto light/dark, in light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path style="opacity: 50%" d="M 5.411 14.504 C 5.471 14.504 5.532 14.504 5.591 14.504 C 3.639 16.319 4.383 19.569 6.931 20.352 C 7.693 20.586 8.512 20.551 9.25 20.252 C 8.023 23.207 4.056 23.725 2.11 21.184 C 0.166 18.642 1.702 14.949 4.874 14.536 C 5.051 14.512 5.231 14.5 5.411 14.5 L 5.411 14.504 Z"/>
      <line x1="14.5" y1="3.25" x2="14.5" y2="1.25"/>
      <line x1="14.5" y1="15.85" x2="14.5" y2="17.85"/>
      <line x1="10.044" y1="5.094" x2="8.63" y2="3.68"/>
      <line x1="19" y1="14.05" x2="20.414" y2="15.464"/>
      <line x1="8.2" y1="9.55" x2="6.2" y2="9.55"/>
      <line x1="20.8" y1="9.55" x2="22.8" y2="9.55"/>
      <line x1="10.044" y1="14.006" x2="8.63" y2="15.42"/>
      <line x1="19" y1="5.05" x2="20.414" y2="3.636"/>
      <circle cx="14.5" cy="9.55" r="3.6"/>
    </svg>
  </symbol>
  <symbol id="svg-moon-with-sun" viewBox="0 0 24 24">
    <title>Auto light/dark, in dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path d="M 8.282 7.007 C 8.385 7.007 8.494 7.007 8.595 7.007 C 5.18 10.184 6.481 15.869 10.942 17.24 C 12.275 17.648 13.706 17.589 15 17.066 C 12.851 22.236 5.91 23.143 2.505 18.696 C -0.897 14.249 1.791 7.786 7.342 7.063 C 7.652 7.021 7.965 7 8.282 7 L 8.282 7.007 Z"/>
      <line style="opacity: 50%" x1="18" y1="3.705" x2="18" y2="2.5"/>
      <line style="opacity: 50%" x1="18" y1="11.295" x2="18" y2="12.5"/>
      <line style="opacity: 50%" x1="15.316" y1="4.816" x2="14.464" y2="3.964"/>
      <line style="opacity: 50%" x1="20.711" y1="10.212" x2="21.563" y2="11.063"/>
      <line style="opacity: 50%" x1="14.205" y1="7.5" x2="13.001" y2="7.5"/>
      <line style="opacity: 50%" x1="21.795" y1="7.5" x2="23" y2="7.5"/>
      <line style="opacity: 50%" x1="15.316" y1="10.184" x2="14.464" y2="11.036"/>
      <line style="opacity: 50%" x1="20.711" y1="4.789" x2="21.563" y2="3.937"/>
      <circle style="opacity: 50%" cx="18" cy="7.5" r="2.169"/>
    </svg>
  </symbol>
  <symbol id="svg-pencil" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-pencil-code">
      <path d="M4 20h4l10.5 -10.5a2.828 2.828 0 1 0 -4 -4l-10.5 10.5v4" />
      <path d="M13.5 6.5l4 4" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
  <symbol id="svg-eye" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-eye-code">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M10 12a2 2 0 1 0 4 0a2 2 0 0 0 -4 0" />
      <path
        d="M11.11 17.958c-3.209 -.307 -5.91 -2.293 -8.11 -5.958c2.4 -4 5.4 -6 9 -6c3.6 0 6.6 2 9 6c-.21 .352 -.427 .688 -.647 1.008" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle site navigation sidebar">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc" aria-label="Toggle table of contents sidebar">
<label class="overlay sidebar-overlay" for="__navigation"></label>
<label class="overlay toc-overlay" for="__toc"></label>

<a class="skip-to-content muted-link" href="#furo-main-content">Skip to content</a>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <span class="icon"><svg><use href="#svg-menu"></use></svg></span>
      </label>
    </div>
    <div class="header-center">
      <a href="../../index.html"><div class="brand">art_modern 1.3.4 documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle" aria-label="Toggle Light / Dark / Auto color theme">
          <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
          <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <span class="icon"><svg><use href="#svg-toc"></use></svg></span>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../../index.html">
  
  <span class="sidebar-brand-text">art_modern 1.3.4 documentation</span>
  
</a><form class="sidebar-search-container" method="get" action="../../search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../README_NEW.html"><code class="docutils literal notranslate"><span class="pre">art_modern</span></code>: Modernized ART Simulator of Diverse Next-Generation Sequencing Reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="Install.html">Installing <code class="docutils literal notranslate"><span class="pre">art_modern</span></code></a></li>
<li class="toctree-l1 current current-page"><a class="current reference internal" href="#">Usage of <code class="docutils literal notranslate"><span class="pre">art_modern</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="Usage-apb.html">Usage of <code class="docutils literal notranslate"><span class="pre">art_profile_builder</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="Usage-amc.html">Usage of <code class="docutils literal notranslate"><span class="pre">am_compress</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="Guide.html">Guide on Miscellaneous Topics</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">Frequently Asked Questions (FAQs)</a></li>
<li class="toctree-l1"><a class="reference internal" href="Design.html">Design Topics</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="Contributing.html">Developing &amp; Contributing</a><input aria-label="Toggle navigation of Developing &amp; Contributing" class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><span class="icon"><svg><use href="#svg-arrow-right"></use></svg></span></label><ul>
<li class="toctree-l2"><a class="reference internal" href="MakefileTargets.html">Toplevel Makefile Documentation</a></li>
<li class="toctree-l2"><a class="reference internal" href="TODO.html">TODO</a></li>
<li class="toctree-l2"><a class="reference internal" href="CODE_OF_CONDUCT.html">Code of Conduct</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Security.html">Security Policy</a></li>
<li class="toctree-l1"><a class="reference internal" href="News.html">News &amp; Release Notes</a></li>
<li class="toctree-l1"><a class="reference internal" href="Copying.html">Copyright</a></li>
<li class="toctree-l1"><a class="reference internal" href="../License.html">GNU GENERAL PUBLIC LICENSE</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          <div class="view-this-page">
  <a class="muted-link" href="../../_sources/src/docs/Usage-art_modern.md.txt" title="View this page">
    <svg><use href="#svg-eye"></use></svg>
    <span class="visually-hidden">View this page</span>
  </a>
</div>
<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle" aria-label="Toggle Light / Dark / Auto color theme">
              <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
              <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <span class="icon"><svg><use href="#svg-toc"></use></svg></span>
          </label>
        </div>
        <article role="main" id="furo-main-content">
          <section class="tex2jax_ignore mathjax_ignore" id="usage-of-art-modern">
<h1>Usage of <code class="docutils literal notranslate"><span class="pre">art_modern</span></code><a class="headerlink" href="#usage-of-art-modern" title="Link to this heading">¶</a></h1>
<p>This is the detailed usage of <code class="docutils literal notranslate"><span class="pre">art_modern</span></code>. Every parameter and their combinations are introduced in detail. In this documentation, commands will be represented as <code class="docutils literal notranslate"><span class="pre">ls</span> <span class="pre">-lFh</span></code> with in-line or block omission represented as <code class="docutils literal notranslate"><span class="pre">[...]</span></code>.</p>
<section id="parameter-controlling-simulation-behavior">
<h2>Parameter Controlling Simulation Behavior<a class="headerlink" href="#parameter-controlling-simulation-behavior" title="Link to this heading">¶</a></h2>
<section id="simulation-modes-mode">
<h3>Simulation Modes (<code class="docutils literal notranslate"><span class="pre">--mode</span></code>)<a class="headerlink" href="#simulation-modes-mode" title="Link to this heading">¶</a></h3>
<p>The simulator currently supports 3 simulation modes:</p>
<ul class="simple">
<li><p><strong><code class="docutils literal notranslate"><span class="pre">wgs</span></code> (DEFAULT) for whole-genome sequencing.</strong> For scenarios with constant coverage over a limited number of sized contigs.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">trans</span></code> for transcriptome simulation. For scenarios with many short contigs and/or contig-specific coverage.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">template</span></code> for template simulation. Often being used with a high-level simulator that produces read fragments.</p></li>
</ul>
</section>
<section id="library-construction-methods-lc">
<h3>Library Construction Methods (<code class="docutils literal notranslate"><span class="pre">--lc</span></code>)<a class="headerlink" href="#library-construction-methods-lc" title="Link to this heading">¶</a></h3>
<p>The simulator currently supports 3 library construction methods:</p>
<section id="se-default-for-single-end-reads">
<h4><strong><code class="docutils literal notranslate"><span class="pre">se</span></code> (DEFAULT) for single-end reads</strong><a class="headerlink" href="#se-default-for-single-end-reads" title="Link to this heading">¶</a></h4>
<ul class="simple">
<li><p><strong>For <code class="docutils literal notranslate"><span class="pre">wgs</span></code> and <code class="docutils literal notranslate"><span class="pre">trans</span></code> mode:</strong> A read may start from any position in the reference contig that is long enough to contain the read.</p></li>
<li><p><strong>For <code class="docutils literal notranslate"><span class="pre">template</span></code> mode:</strong> A read will start from position 0 in the reference contig.</p></li>
</ul>
<p>A diagram for single-end read extraction:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>FRAGMENT:  |==============================|
SE READ:   |-------&gt;
</pre></div>
</div>
</section>
<section id="pe-for-paired-end-reads">
<h4><code class="docutils literal notranslate"><span class="pre">pe</span></code> for paired-end reads<a class="headerlink" href="#pe-for-paired-end-reads" title="Link to this heading">¶</a></h4>
<ul class="simple">
<li><p><strong>For <code class="docutils literal notranslate"><span class="pre">wgs</span></code> and <code class="docutils literal notranslate"><span class="pre">trans</span></code> mode:</strong> A fragment (called insert in some other simulators), whose length is drawn from a Gaussian distribution specified by <code class="docutils literal notranslate"><span class="pre">--pe_frag_dist_mean</span></code> and <code class="docutils literal notranslate"><span class="pre">--pe_frag_dist_std_dev</span></code>, is created from any position in the reference contig that is long enough to contain the fragment.</p></li>
<li><p><strong>For <code class="docutils literal notranslate"><span class="pre">template</span></code> mode:</strong> A fragment that spans the entire length of the reference contig is created.</p></li>
<li><p>Paired-end reads facing inward of the fragment are then created from both ends of the fragment.</p></li>
</ul>
<p>A diagram for paired-end read extraction:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>FRAGMENT:  |==============================|
PE READ 1: |-------&gt;
PE READ 2:                        |&lt;------|
</pre></div>
</div>
</section>
<section id="mp-for-mate-paired-reads">
<h4><code class="docutils literal notranslate"><span class="pre">mp</span></code> for mate-paired reads<a class="headerlink" href="#mp-for-mate-paired-reads" title="Link to this heading">¶</a></h4>
<ul class="simple">
<li><p>Extraction of the fragment as-is in <code class="docutils literal notranslate"><span class="pre">pe</span></code> mode.</p></li>
<li><p>Mate-paired reads facing outward of the fragment are created from both ends of the fragment.</p></li>
</ul>
<p>A diagram for mate-pair extraction:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>FRAGMENT:  |==============================|
MP READ 1:                        |-------&gt;
MP READ 2: &lt;-------|
</pre></div>
</div>
</section>
<section id="see-also">
<h4>See also<a class="headerlink" href="#see-also" title="Link to this heading">¶</a></h4>
<ul class="simple">
<li><p><a class="reference external" href="https://www.illumina.com/science/technology/next-generation-sequencing/plan-experiments/paired-end-vs-single-read.html">Illumina paired vs. single end blog</a>.</p></li>
<li><p><a class="reference external" href="https://www.illumina.com/science/technology/next-generation-sequencing/mate-pair-sequencing.html">Illumina explaination of mate-pair simulation</a>.</p></li>
</ul>
</section>
</section>
<section id="thread-level-parallelism-parallel">
<span id="parallelism-section"></span><h3>Thread-Level Parallelism (<code class="docutils literal notranslate"><span class="pre">--parallel</span></code>)<a class="headerlink" href="#thread-level-parallelism-parallel" title="Link to this heading">¶</a></h3>
<p>Number of threads to use. Use <code class="docutils literal notranslate"><span class="pre">-1</span></code> to disable parallelism. Use <code class="docutils literal notranslate"><span class="pre">0</span></code> (default) to use all available cores. Oversubscription is allowed but not recommended.</p>
<p><strong>NOTE</strong> If MPI is enabled, each MPI process will use the specified number of threads. For example, if you run <code class="docutils literal notranslate"><span class="pre">mpiexec</span> <span class="pre">-n</span> <span class="pre">4</span> <span class="pre">art_modern</span> <span class="pre">--parallel</span> <span class="pre">2</span> <span class="pre">...</span></code>, there will be 8 threads in total.</p>
<p><strong>NOTE</strong> SAM/BAM output writer is parallelized using its own thread pool. See parameter <a class="reference internal" href="#out-sam-section"><code class="docutils literal notranslate"><span class="pre">--o-sam-num_threads</span></code></a> and <a class="reference internal" href="#out-hl-sam-section"><code class="docutils literal notranslate"><span class="pre">--o-hl_sam-num_threads</span></code></a> for more information. For example, if you run <code class="docutils literal notranslate"><span class="pre">art_modern</span> <span class="pre">--parallel</span> <span class="pre">4</span> <span class="pre">--o-sam-num_threads</span> <span class="pre">2</span> <span class="pre">--o-hl_sam-num_threads</span> <span class="pre">3</span> <span class="pre">...</span></code>, there will be 9 threads in total.</p>
<p><strong>WARNING</strong> Do not oversubscribe the number of threads. It impacts the performance of the program.</p>
</section>
<section id="error-profile-related-parameters">
<h3>Error-Profile Related Parameters<a class="headerlink" href="#error-profile-related-parameters" title="Link to this heading">¶</a></h3>
<section id="which-quality-distribution-file-to-use">
<h4>Which Quality Distribution File to Use<a class="headerlink" href="#which-quality-distribution-file-to-use" title="Link to this heading">¶</a></h4>
<p>Quality distribution files specify the base quality distribution at each position of the read. They are used to simulate sequencing errors.</p>
<p>Users may supply self-trained quality distribution files using <code class="docutils literal notranslate"><span class="pre">--qual_file_1</span></code> and <code class="docutils literal notranslate"><span class="pre">--qual_file_2</span></code> for read 1 and read 2 respectively. The format of quality distribution files is compatible with the original ART. The first parameter is required while the second parameter is required only for <code class="docutils literal notranslate"><span class="pre">pe</span></code> and <code class="docutils literal notranslate"><span class="pre">mp</span></code> library construction mode.</p>
<p>Users may also use built-in quality distribution files using <code class="docutils literal notranslate"><span class="pre">--builtin_qual_file</span></code> (added in <a class="reference internal" href="News.html#v-1-1-0-section"><span class="std std-ref">1.1.0</span></a>.). The built-in quality distribution files are bundled with the simulator. Note that the built-in quality distribution files comes with different maximum read-length and different support over paired-end reads. So please make sure that the read length specified by <code class="docutils literal notranslate"><span class="pre">--read_len</span></code> is compatible with the built-in quality distribution file.</p>
<p>Quality distributions bundled with the original ART can be found <a class="reference external" href="https://github.com/YU-Zhejian/art_modern/tree/master/data/Illumina_profiles">here</a>. In details:</p>
<p>Here lists existing Illumina models supported by the original ART, which is still available in <code class="docutils literal notranslate"><span class="pre">art_modern</span></code>.</p>
<div class="table-wrapper colwidths-auto docutils container">
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Name</p></th>
<th class="head"><p>Name (<code class="docutils literal notranslate"><span class="pre">art_modern</span></code>)</p></th>
<th class="head"><p>R1 <code class="docutils literal notranslate"><span class="pre">txt</span></code> file name</p></th>
<th class="head"><p>R2?</p></th>
<th class="head"><p>RLen1</p></th>
<th class="head"><p>RLen2</p></th>
<th class="head"><p>Model Name</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">GA1</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">GA1Recalibrated_36bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">EmpR36R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>36</p></td>
<td><p>36</p></td>
<td><p>GenomeAnalyzer I</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">GA1</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">GA1Recalibrated_44bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">EmpR44R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>44</p></td>
<td><p>44</p></td>
<td><p>GenomeAnalyzer I</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">GA2</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">GA2Recalibrated_50bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">EmpR50R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>50</p></td>
<td><p>50</p></td>
<td><p>GenomeAnalyzer II</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">GA2</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">GA2Recalibrated_75bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">EmpR75R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>75</p></td>
<td><p>75</p></td>
<td><p>GenomeAnalyzer II</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">HS10</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">HiSeq1000_100bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">Emp100R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>HiSeq 1000</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">MSv1</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">MiSeq_250bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">EmpMiSeq250R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>250</p></td>
<td><p>250</p></td>
<td><p>MiSeq</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">HS20</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">HiSeq2000_100bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">HiSeq2kL100R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>100</p></td>
<td><p>100</p></td>
<td><p>HiSeq 2000</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">HS25</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">HiSeq2500_125bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">HiSeq2500L125R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>126</p></td>
<td><p>126</p></td>
<td><p>HiSeq 2500</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">HS25</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">HiSeq2500_150bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">HiSeq2500L150R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>150</p></td>
<td><p>150</p></td>
<td><p>HiSeq 2500</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">HSXn</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">HiSeqX_PCR_Free_150bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">HiSeqXPCRfreeL150R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>151</p></td>
<td><p>151</p></td>
<td><p>HiSeqX PCR free</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">HSXt</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">HiSeqX_TruSeq_150bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">HiSeqXtruSeqL150R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>151</p></td>
<td><p>151</p></td>
<td><p>HiSeqX TruSeq</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">MinS</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">MiniSeq_TruSeq_50bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">MiniSeqTruSeqL50</span></code></p></td>
<td><p>N</p></td>
<td><p>51</p></td>
<td><p>N/A</p></td>
<td><p>MiniSeq TruSeq</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">MSv3</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">MiSeq_v3_250bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">MiSeqv3L250R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>251</p></td>
<td><p>251</p></td>
<td><p>MiSeq v3</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">NS50</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">NextSeq500_v2_75bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">NextSeq500v2L75R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>76</p></td>
<td><p>76</p></td>
<td><p>NextSeq500 v2</p></td>
</tr>
</tbody>
</table>
</div>
<p>Following profiles are only available in <code class="docutils literal notranslate"><span class="pre">art_modern</span></code> but not in the original ART as a built-in model. However, the
model error files are provided in the original ART.</p>
<div class="table-wrapper colwidths-auto docutils container">
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Name (<code class="docutils literal notranslate"><span class="pre">art_modern</span></code>)</p></th>
<th class="head"><p>R1 <code class="docutils literal notranslate"><span class="pre">txt</span></code> file name</p></th>
<th class="head"><p>R2?</p></th>
<th class="head"><p>RLen1</p></th>
<th class="head"><p>RLen2</p></th>
<th class="head"><p>Model Name</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">GA1_36bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">Emp36R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>36</p></td>
<td><p>36</p></td>
<td><p>GenomeAnalyzer I</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">GA1_44bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">Emp44R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>44</p></td>
<td><p>44</p></td>
<td><p>GenomeAnalyzer I</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">GA2_50bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">Emp50R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>53</p></td>
<td><p>50</p></td>
<td><p>GenomeAnalyzer II</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">GA2_75bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">Emp75R1</span></code></p></td>
<td><p>Y</p></td>
<td><p>75</p></td>
<td><p>75</p></td>
<td><p>GenomeAnalyzer II</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">HiSeq2500Filtered_150bp</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">HiSeq2500L150R1filter</span></code></p></td>
<td><p>Y</p></td>
<td><p>150</p></td>
<td><p>150</p></td>
<td><p>HiSeq 2500</p></td>
</tr>
</tbody>
</table>
</div>
</section>
<section id="shift-and-clip-parameters">
<h4>Shift and Clip Parameters<a class="headerlink" href="#shift-and-clip-parameters" title="Link to this heading">¶</a></h4>
<p><code class="docutils literal notranslate"><span class="pre">--q_shift_1</span></code> and <code class="docutils literal notranslate"><span class="pre">--q_shift_2</span></code>: Shift the base quality of the quality distribution by the specified value. Please note that the shifted quality distribution will be bounded to <code class="docutils literal notranslate"><span class="pre">--min_qual</span></code> and <code class="docutils literal notranslate"><span class="pre">--max_qual</span></code>.</p>
<p><code class="docutils literal notranslate"><span class="pre">--min_qual</span></code> and <code class="docutils literal notranslate"><span class="pre">--max_qual</span></code>: Clip the quality distribution to the specified range.</p>
</section>
<section id="separate-quality-distributions-for-different-bases-sep-flag">
<h4>Separate Quality Distributions for Different Bases (<code class="docutils literal notranslate"><span class="pre">--sep_flag</span></code>)<a class="headerlink" href="#separate-quality-distributions-for-different-bases-sep-flag" title="Link to this heading">¶</a></h4>
<p>This flag can separate quality distributions for different bases in the same position.</p>
<ul class="simple">
<li><p><strong>unset (DEFAULT)</strong>: Use the same quality distribution for different bases in the same position. For example, the quality distribution of pos 10 will be the same regardless whether the incoming base is <code class="docutils literal notranslate"><span class="pre">A</span></code> or <code class="docutils literal notranslate"><span class="pre">T</span></code>.</p></li>
<li><p>set: Use different quality distributions for different bases in the same position. For example, the quality distribution of pos 10 may <strong>NOT</strong> be the same regardless whether the incoming base is <code class="docutils literal notranslate"><span class="pre">A</span></code> or <code class="docutils literal notranslate"><span class="pre">T</span></code>.</p></li>
</ul>
</section>
</section>
</section>
<section id="parameters-controlling-input">
<h2>Parameters Controlling Input<a class="headerlink" href="#parameters-controlling-input" title="Link to this heading">¶</a></h2>
<section id="input-file-path-i-file">
<h3>Input File Path (<code class="docutils literal notranslate"><span class="pre">--i-file</span></code>)<a class="headerlink" href="#input-file-path-i-file" title="Link to this heading">¶</a></h3>
<p>The input reference file path. It must exist in the filesystem. Device paths like <code class="docutils literal notranslate"><span class="pre">/dev/stdin</span></code> is supported for non-MPI builds.</p>
<p>Currently, we support input in FASTA and PBSIM3 Transcripts format. An additional parameter <a class="reference internal" href="#input-file-type-section"><code class="docutils literal notranslate"><span class="pre">--i-type</span></code></a> is used to specify the file type.</p>
<p><strong>NOTE</strong> Some parser may support compressed FASTAs. See <a class="reference internal" href="#input-parser-section"><code class="docutils literal notranslate"><span class="pre">--i-parser</span></code></a> for more detils.</p>
</section>
<section id="input-file-type-i-type">
<span id="input-file-type-section"></span><h3>Input File Type (<code class="docutils literal notranslate"><span class="pre">--i-type</span></code>)<a class="headerlink" href="#input-file-type-i-type" title="Link to this heading">¶</a></h3>
<p>The file type of input reference sequences. Currently, FASTA and PBSIM3 Transcripts format are supported.</p>
<ul class="simple">
<li><p><strong><code class="docutils literal notranslate"><span class="pre">auto</span></code> (DEFAULT) for extension-based decision.</strong></p>
<ul>
<li><p>If the file ends with <code class="docutils literal notranslate"><span class="pre">.fna</span></code>, <code class="docutils literal notranslate"><span class="pre">.fsa</span></code>, <code class="docutils literal notranslate"><span class="pre">.fa</span></code>, <code class="docutils literal notranslate"><span class="pre">.fasta</span></code>, resolve to <code class="docutils literal notranslate"><span class="pre">fasta</span></code>.</p></li>
<li><p>Otherwise, an error will be raised.</p></li>
</ul>
</li>
<li><p><code class="docutils literal notranslate"><span class="pre">fasta</span></code> for FASTA files.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">pbsim3_transcripts</span></code> for PBSIM3 Transcripts format.</p></li>
</ul>
<p><strong>NOTE</strong> If you’re using UNIX devices like <code class="docutils literal notranslate"><span class="pre">/dev/stdin</span></code> as input, please make sure that this option is specified.</p>
</section>
<section id="input-parser-i-parser">
<span id="input-parser-section"></span><h3>Input Parser (<code class="docutils literal notranslate"><span class="pre">--i-parser</span></code>)<a class="headerlink" href="#input-parser-i-parser" title="Link to this heading">¶</a></h3>
<section id="auto-default-for-size-based-determination">
<h4><strong><code class="docutils literal notranslate"><span class="pre">auto</span></code> (DEFAULT) for size-based determination.</strong><a class="headerlink" href="#auto-default-for-size-based-determination" title="Link to this heading">¶</a></h4>
<p>If the file size larger than 1 GiB or cannot be told (which is quite common if the input was redirected from stdin or other devices), resolve to <code class="docutils literal notranslate"><span class="pre">htslib</span></code> (<code class="docutils literal notranslate"><span class="pre">wgs</span></code> mode) or <code class="docutils literal notranslate"><span class="pre">stream</span></code> (<code class="docutils literal notranslate"><span class="pre">trans</span></code> or <code class="docutils literal notranslate"><span class="pre">template</span></code> mode). Otherwise, use <code class="docutils literal notranslate"><span class="pre">memory</span></code>.</p>
</section>
<section id="memory-for-in-memory-reference-parser">
<h4><code class="docutils literal notranslate"><span class="pre">memory</span></code> for in-memory reference parser<a class="headerlink" href="#memory-for-in-memory-reference-parser" title="Link to this heading">¶</a></h4>
<p>The entire file will be read into the memory. Fast for small reference files.</p>
</section>
<section id="htslib-for-htslib-faidx-parser">
<h4><code class="docutils literal notranslate"><span class="pre">htslib</span></code> for HTSLib FAIDX parser<a class="headerlink" href="#htslib-for-htslib-faidx-parser" title="Link to this heading">¶</a></h4>
<p>Store FASTA Index in memory and fetch sequences from disk using <a class="reference external" href="https://github.com/samtools/htslib">HTSLib</a>. Memory-efficient for large genome assembly FASTA files with limited number of long contigs, but inefficient for transcriptome/template FASTAs with large number of (relatively) short contigs.</p>
<p>The FASTA index, usually ends with <code class="docutils literal notranslate"><span class="pre">.fai</span></code> and can be built using <code class="docutils literal notranslate"><span class="pre">samtools</span> <span class="pre">faidx</span></code>, will be created automatically if not exist.</p>
<p><strong>NOTE</strong> For <code class="docutils literal notranslate"><span class="pre">wgs</span></code> mode only.</p>
<p><strong>NOTE</strong> Do not support PBSIM3 Transcripts format.</p>
<p><strong>NOTE</strong> Do not support UNIX devices for input.</p>
<p><strong>NOTE</strong> <code class="docutils literal notranslate"><span class="pre">htslib</span></code> parser can be used on a <code class="docutils literal notranslate"><span class="pre">.fa.gz</span></code> file. However, please note that:</p>
<ul>
<li><p>The GZip file <strong>MUST</strong> be <a class="reference external" href="https://www.htslib.org/doc/bgzip.html"><code class="docutils literal notranslate"><span class="pre">bgzip</span></code></a>-compressed. Plain GZip will <strong>NOT</strong> work. If you try to use a GZip-compressed FASTA file, you’ll see error messages like:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>[E::fai_build_core] File truncated at line 1
[E::fai_build3_core] Cannot index files compressed with gzip, please use bgzip
[E::fai_path] Failed to build index file for reference file [...]
</pre></div>
</div>
</li>
<li><p>Specify <code class="docutils literal notranslate"><span class="pre">--i-type</span> <span class="pre">fasta</span></code> to use this parser.</p></li>
</ul>
</section>
<section id="stream-for-one-pass-streaming-parser">
<h4><code class="docutils literal notranslate"><span class="pre">stream</span></code> for one-pass streaming parser<a class="headerlink" href="#stream-for-one-pass-streaming-parser" title="Link to this heading">¶</a></h4>
<p>Streamline the input reference as batches and process them one by one. Efficient for transcriptome/template FASTAs with large number of (relatively) short contigs.</p>
<p>Please note that here, the contig number and length are not accessible to the output dispatcher. So use <a class="reference internal" href="#out-hl-sam-section">headless SAM/BAM output</a> if a SAM/BAM output is needed.</p>
<p>The batch size of each batch is controlled by <code class="docutils literal notranslate"><span class="pre">--i-batch_size</span></code>. Larger batch size may improve performance but requires more memory. Use larger batch size if the targeted sequencing depth is relatively shallow.</p>
</section>
</section>
<section id="coverage-i-fcov">
<h3>Coverage (<code class="docutils literal notranslate"><span class="pre">--i-fcov</span></code>)<a class="headerlink" href="#coverage-i-fcov" title="Link to this heading">¶</a></h3>
<p>This option allows you to specify coverage. <code class="docutils literal notranslate"><span class="pre">art_modern</span></code> supports the following coverage mode:</p>
<ul class="simple">
<li><p>Unified coverage floating point in <code class="docutils literal notranslate"><span class="pre">double</span></code> data type (e.g., <code class="docutils literal notranslate"><span class="pre">10.0</span></code>). Under this scenario, the coverage is identical for all contigs.</p></li>
<li><p>Per-contig coverage headless tab-separated value (TSV) without strand information.</p></li>
<li><p>Per-contig coverage headless TSV with strand information.</p></li>
</ul>
<p><strong>NOTE</strong> This parameter is ignored if the file type is set to <code class="docutils literal notranslate"><span class="pre">pbsim3_transcripts</span></code>. This format comes with contig-specific coverage information.</p>
<p><strong>NOTE</strong> We prefer unified coverage floating points to TSVs. That is, if you set this parameter to 10.0, we’re <strong>NOT</strong> going to check whether there’s a file named <code class="docutils literal notranslate"><span class="pre">10.0</span></code>. So please make sure that the file name is not a number if you want to specify a coverage file.</p>
<p><strong>NOTE</strong> For coverage of the template mode:</p>
<ul class="simple">
<li><p>If a unified coverage file is provided, the coverage will be intepreted as positive coverage. That is:</p>
<ul>
<li><p>If the library is SE, all reads will be generated from the positive strand at the 1st base.</p></li>
<li><p>If the library is PE, all read 1 will be generated from the positive strand at the first base, with read 2 generated from the negative strand at the last base.</p></li>
<li><p>If the library is MP, all read 1 will be generated from the positive strand to the last base, with read 2 generated from the negative strand to the first base.</p></li>
</ul>
</li>
<li><p>If a 2-column (unstranded) coverage file is provided, the coverage will be intepreted as positive coverage.</p>
<ul>
<li><p>Introduced in <a class="reference internal" href="News.html#v-1-3-0-section"><span class="std std-ref">1.3.0</span></a>.</p></li>
</ul>
</li>
<li><p>If a 3-column (stranded) coverage file or input in format of <code class="docutils literal notranslate"><span class="pre">pvsim3_transcripts</span></code> is provided, the coverage will be intepreted as-is.</p></li>
</ul>
<p>For example, if you specified <code class="docutils literal notranslate"><span class="pre">--read_len_1</span> <span class="pre">10</span> <span class="pre">--read_len_2</span> <span class="pre">150</span></code> with unified/unstranded coverage 5 and PE library construction mode, you’ll be likely to see:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>|================================================|
|----&gt;                              &lt;------------|
|----&gt;                              &lt;------------|
|----&gt;                              &lt;------------|
|----&gt;                              &lt;------------|
|----&gt;                              &lt;------------|
</pre></div>
</div>
<p>With stranded coverage positive=3 and negative=2, you may see:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>|================================================|
|----&gt;                              &lt;------------|
|----&gt;                              &lt;------------|
|----&gt;                              &lt;------------|
|------------&gt;                              &lt;----|
|------------&gt;                              &lt;----|
</pre></div>
</div>
<p>while for MP library construction mode you would see:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>|================================================|
&lt;------------|                              |----&gt;
&lt;------------|                              |----&gt;
&lt;------------|                              |----&gt;
&lt;----|                              |------------&gt;
&lt;----|                              |------------&gt;
</pre></div>
</div>
</section>
<section id="compatibility-matrices-of-input-parameters">
<h3>Compatibility Matrices of Input Parameters<a class="headerlink" href="#compatibility-matrices-of-input-parameters" title="Link to this heading">¶</a></h3>
<p>Compatibility matrix of file type, simulation mode, and parser:</p>
<div class="table-wrapper colwidths-auto docutils container">
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Parser \ Mode</p></th>
<th class="head"><p><code class="docutils literal notranslate"><span class="pre">wgs</span></code></p></th>
<th class="head"><p><code class="docutils literal notranslate"><span class="pre">trans</span></code></p></th>
<th class="head"><p><code class="docutils literal notranslate"><span class="pre">template</span></code></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">memory</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">fasta</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">fasta</span></code> / <code class="docutils literal notranslate"><span class="pre">pbsim3_transcripts</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">fasta</span></code> / <code class="docutils literal notranslate"><span class="pre">pbsim3_transcripts</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">htslib</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">fasta</span></code></p></td>
<td><p><strong>ERROR</strong></p></td>
<td><p><strong>ERROR</strong></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">stream</span></code></p></td>
<td><p><strong>ERROR</strong></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">fasta</span></code> / <code class="docutils literal notranslate"><span class="pre">pbsim3_transcripts</span></code></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">fasta</span></code> / <code class="docutils literal notranslate"><span class="pre">pbsim3_transcripts</span></code></p></td>
</tr>
</tbody>
</table>
</div>
<p>Compatibility matrix of coverage mode, simulation mode, and file type:</p>
<div class="table-wrapper colwidths-auto docutils container">
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Mode \ File Type</p></th>
<th class="head"><p><code class="docutils literal notranslate"><span class="pre">fasta</span></code></p></th>
<th class="head"><p><code class="docutils literal notranslate"><span class="pre">pbsim3_transcripts</span></code></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">wgs</span></code></p></td>
<td><p>Unified</p></td>
<td><p><strong>ERROR</strong></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">trans</span></code></p></td>
<td><p>Unified / Unstraded / Stranded</p></td>
<td><p><strong>IGNORED</strong></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">template</span></code></p></td>
<td><p>Unified / Unstraded / Stranded</p></td>
<td><p><strong>IGNORED</strong></p></td>
</tr>
</tbody>
</table>
</div>
</section>
</section>
<section id="parameters-controlling-output">
<h2>Parameters Controlling Output<a class="headerlink" href="#parameters-controlling-output" title="Link to this heading">¶</a></h2>
<p>Here introduces diverse output formats supported by <code class="docutils literal notranslate"><span class="pre">art_modern</span></code>. You may specify none of them to perform simulation without any output for benchmarking purposes.</p>
<p>If the parent directory does not exist, it will be created using a <code class="docutils literal notranslate"><span class="pre">mkdir</span> <span class="pre">-p</span></code>-like manner.</p>
<p>For all output format, a queue size parameter (e.g., <code class="docutils literal notranslate"><span class="pre">--o-pwa-queue_size</span></code>) is provided to control the size of lockfree queue. Turn this up to improve performance when you have enough memory, and low if the application is consuming too much memory. Added in <a class="reference internal" href="News.html#v-1-3-2-section"><span class="std std-ref">1.3.2</span></a>.</p>
<section id="pairwise-alignment-format-o-pwa">
<h3>Pairwise Alignment Format (<code class="docutils literal notranslate"><span class="pre">--o-pwa</span></code>)<a class="headerlink" href="#pairwise-alignment-format-o-pwa" title="Link to this heading">¶</a></h3>
<p>PWA is serialization of the internally used data structure of the simulator. Currently, PWA consists of some metadata lines (Started with <code class="docutils literal notranslate"><span class="pre">#</span></code>) and a list of alignments, each spanning 4 lines.</p>
<p>The alignment lines are as follows:</p>
<ol class="arabic simple">
<li><p><code class="docutils literal notranslate"><span class="pre">&gt;</span></code>, read name, a blank, and the leftmost position of the alignment.</p></li>
<li><p>The aligned read sequence, gapped with <code class="docutils literal notranslate"><span class="pre">-</span></code>.</p></li>
<li><p>The aligned reference sequence, gapped with <code class="docutils literal notranslate"><span class="pre">-</span></code>.</p></li>
<li><p>The gapless quality string.</p></li>
</ol>
<p>Example:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>#PWA
#ARGS: opt/build_debug/art_modern [...]
&gt;NM_069135:art_modern:1:nompi:0	NM_069135:0:+
AGC-CAAACGGGCAACCAGACTCCGC[...]
AGCACAAA-GGGCAACCAGACTCCGC[...]
BCCCCGGGGGGGFGGGGGGGFGGGGG[...]
[...]
</pre></div>
</div>
<p>This output format supports <a class="reference internal" href="Usage-amc.html#am-compress-usage-section"><span class="std std-ref"><code class="docutils literal notranslate"><span class="pre">am_compress</span></code></span></a>-like compression arguments (Namely, <code class="docutils literal notranslate"><span class="pre">--o-pwa-compression</span></code>, <code class="docutils literal notranslate"><span class="pre">--o-pwa-compression_level</span></code>, <code class="docutils literal notranslate"><span class="pre">--o-pwa-buffer_size</span></code> and <code class="docutils literal notranslate"><span class="pre">--o-pwa-num_threads</span></code>). Added in <a class="reference internal" href="News.html#v-1-3-4-section"><span class="std std-ref">1.3.4</span></a>.</p>
</section>
<section id="fastq-format-o-fastq">
<h3>FASTQ Format (<code class="docutils literal notranslate"><span class="pre">--o-fastq</span></code>)<a class="headerlink" href="#fastq-format-o-fastq" title="Link to this heading">¶</a></h3>
<p>The good old FASTQ format. The qualities are Phread encoded in ASCII with an offset of 33.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>@NM_069135:art_modern:1:nompi:0
AGCCAAACGGGCAACCAGACTCCGCC[...]
+
BCCCCGGGGGGGFGGGGGGGFGGGGG[...]
</pre></div>
</div>
<p>FASTQ files can be easily converted to FASTA using <a class="reference external" href="https://github.com/lh3/seqtk"><code class="docutils literal notranslate"><span class="pre">seqtk</span></code></a>:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cat<span class="w"> </span><span class="k">in</span>.fq<span class="w"> </span><span class="p">|</span><span class="w"> </span>seqtk<span class="w"> </span>seq<span class="w"> </span>-A<span class="w"> </span>&gt;<span class="w"> </span>out.fa
</pre></div>
</div>
<p>For PE/MP simulation: The generated file is interleaved FASTQ. Split to 2 FASTQ files using <a class="reference external" href="https://github.com/lh3/seqtk"><code class="docutils literal notranslate"><span class="pre">seqtk</span></code></a>:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="c1"># Read 1</span>
seqtk<span class="w"> </span>seq<span class="w"> </span>a.fq<span class="w"> </span>-1<span class="w"> </span>&gt;<span class="w"> </span>a_1.fq
<span class="c1"># Read 2</span>
seqtk<span class="w"> </span>seq<span class="w"> </span>a.fq<span class="w"> </span>-2<span class="w"> </span>&gt;<span class="w"> </span>a_2.fq
</pre></div>
</div>
<p>This output format supports <a class="reference internal" href="Usage-amc.html#am-compress-usage-section"><span class="std std-ref"><code class="docutils literal notranslate"><span class="pre">am_compress</span></code></span></a>-like compression arguments. Added in <a class="reference internal" href="News.html#v-1-3-4-section"><span class="std std-ref">1.3.4</span></a>.</p>
<p>See also:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://genome.ucsc.edu/ENCODE/fileFormats.html#FASTQ">Specifications of Common File Formats Used by the ENCODE Consortium at UCSC</a>.</p></li>
<li><p><a class="reference external" href="https://www.encodeproject.org/help/file-formats/#fastq">Common File Formats Used by the ENCODE Consortium</a>.</p></li>
</ul>
</section>
<section id="fasta-format-o-fasta">
<h3>FASTA Format (<code class="docutils literal notranslate"><span class="pre">--o-fasta</span></code>)<a class="headerlink" href="#fasta-format-o-fasta" title="Link to this heading">¶</a></h3>
<p>Sequence-only output format. Example:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>&gt;NM_069135:art_modern:1:nompi:0
AGCCAAACGGGCAACCAGACTCCGCC[...]
</pre></div>
</div>
<p>This simulator generates one-line FASTA files. That is, each sequence occupies only one line.</p>
<p>This output format supports <a class="reference internal" href="Usage-amc.html#am-compress-usage-section"><span class="std std-ref"><code class="docutils literal notranslate"><span class="pre">am_compress</span></code></span></a>-like compression arguments. Added in <a class="reference internal" href="News.html#v-1-3-4-section"><span class="std std-ref">1.3.4</span></a>.</p>
</section>
<section id="sam-bam-format-o-sam">
<span id="out-sam-section"></span><h3>SAM/BAM Format (<code class="docutils literal notranslate"><span class="pre">--o-sam</span></code>)<a class="headerlink" href="#sam-bam-format-o-sam" title="Link to this heading">¶</a></h3>
<p>Sequence Alignment/Map (SAM) and Binary Alignment/Map (BAM) format supports storing of ground-truth alignment information and other miscellaneous parameters. They can be parsed using <a class="reference external" href="https://github.com/samtools/samtools">samtools</a>, <a class="reference external" href="https://pysam.readthedocs.io/"><code class="docutils literal notranslate"><span class="pre">pysam</span></code></a> and other libraries, and can be used as ground-truth when benchmarking sequence aligners.</p>
<p>This writes canonical SAM/BAM format using <a class="reference external" href="https://github.com/samtools/htslib">HTSLib</a>. This output writer supports computing <code class="docutils literal notranslate"><span class="pre">NM</span></code> and <code class="docutils literal notranslate"><span class="pre">MD</span></code> tag. The mapping qualities for all reads are set to 255.</p>
<p>Example (Tabs replaced with spaces):</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>@HD	VN:1.4	SO:unsorted
@PG	ID:01	PN:art_modern	CL:opt/build_debug/art_modern [...]
@SQ	SN:NM_069135	LN:1718
[...]
NM_069135:[...] 0 NM_069135 1 255 3=[...]5= = 1 0 AGCC[...] BCCC[...] MD:[...] NM:[...]
[...]
</pre></div>
</div>
<p>Other SAM/BAM formatting parameters includes:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">--o-sam-use_m</span></code>: Computing CIGAR string using <code class="docutils literal notranslate"><span class="pre">M</span></code> (<code class="docutils literal notranslate"><span class="pre">BAM_CMATCH</span></code>) instead of <code class="docutils literal notranslate"><span class="pre">=</span></code>/<code class="docutils literal notranslate"><span class="pre">X</span></code> (<code class="docutils literal notranslate"><span class="pre">BAM_CEQUAL</span></code>/<code class="docutils literal notranslate"><span class="pre">BAM_CDIFF</span></code>). Rarely used in next-generation sequencing but common for long-read sequencing alignments.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--o-sam-write_bam</span></code>: Write BAM instead of SAM.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--o-sam-num_threads</span></code>: Number of threads used to compress BAM output.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--o-sam-compress_level</span></code>: <a class="reference external" href="https://www.zlib.net/"><code class="docutils literal notranslate"><span class="pre">zlib</span></code></a> compression level. Supports <code class="docutils literal notranslate"><span class="pre">[u0-9]</span></code> with <code class="docutils literal notranslate"><span class="pre">u</span></code> for uncompressed BAM stream, 0 for uncompressed stream with <code class="docutils literal notranslate"><span class="pre">zlib</span></code> wrapping, and 1–9 for fastest to best compression ratio. Defaults to <code class="docutils literal notranslate"><span class="pre">4</span></code>. Please note that both <code class="docutils literal notranslate"><span class="pre">u</span></code> and <code class="docutils literal notranslate"><span class="pre">0</span></code> generates uncompressed output. However, <code class="docutils literal notranslate"><span class="pre">0</span></code> generates BAM stream with <code class="docutils literal notranslate"><span class="pre">zlib</span></code> wrapping while <code class="docutils literal notranslate"><span class="pre">u</span></code> generates raw BAM stream.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--o-sam-without_tag_MD</span></code>: Do not write <code class="docutils literal notranslate"><span class="pre">MD</span></code> tag. Added in <a class="reference internal" href="News.html#v-1-3-3-section"><span class="std std-ref">1.3.3</span></a>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--o-sam-without_tag_NM</span></code>: Do not write <code class="docutils literal notranslate"><span class="pre">NM</span></code> tag. Added in <a class="reference internal" href="News.html#v-1-3-3-section"><span class="std std-ref">1.3.3</span></a>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--o-sam-no_qual</span></code>: Do not write quality string (Write quality string as <code class="docutils literal notranslate"><span class="pre">*</span></code>). Added in <a class="reference internal" href="News.html#v-1-3-3-section"><span class="std std-ref">1.3.3</span></a>.</p></li>
</ul>
<p><strong>NOTE</strong> For PE/MP simulation, the 2 reads from one fragment may <strong>NOT</strong> appear together.</p>
<p>See also: <a class="reference external" href="https://samtools.github.io/hts-specs/SAMv1.pdf"><code class="docutils literal notranslate"><span class="pre">SAMv1.pdf</span></code></a> for more information on SAM/BAM format.</p>
</section>
<section id="headless-sam-bam-format-o-hl-sam">
<span id="out-hl-sam-section"></span><h3>Headless SAM/BAM Format (<code class="docutils literal notranslate"><span class="pre">--o-hl_sam</span></code>)<a class="headerlink" href="#headless-sam-bam-format-o-hl-sam" title="Link to this heading">¶</a></h3>
<p>This format is specially designed for the <code class="docutils literal notranslate"><span class="pre">stream</span></code> reference parser that allows handling of numerous contigs. As a side effect, the contig name and length information will not be written to SAM header. Each read will be written as unaligned with coordinate information populated in the <code class="docutils literal notranslate"><span class="pre">OA</span></code> tag.</p>
<p>Example (Tabs replaced with spaces):</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>@HD	VN:1.4	SO:unsorted
@PG	ID:01	PN:art_modern	CL:opt/build_debug/art_modern [...]
NM_069135:[...] 4 * 1 0 * * 1 0 AGCC[...] BCCC[...] OA:[...] MD:[...] NM:[...]
[...]
</pre></div>
</div>
<p>An additional option: <code class="docutils literal notranslate"><span class="pre">--o-hl_sam-without_tag_OA</span></code>, setting such which will not write <code class="docutils literal notranslate"><span class="pre">OA</span></code> tag. Added in <a class="reference internal" href="News.html#v-1-3-3-section"><span class="std std-ref">1.3.3</span></a>.</p>
<p>This output writer supports other BAM formatting parameters.</p>
</section>
</section>
<section id="miscellaneous-parameters">
<h2>Miscellaneous Parameters<a class="headerlink" href="#miscellaneous-parameters" title="Link to this heading">¶</a></h2>
<section id="read-length-parameters">
<h3>Read Length Parameters<a class="headerlink" href="#read-length-parameters" title="Link to this heading">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">--read_len</span></code>, <code class="docutils literal notranslate"><span class="pre">--read_len_1</span></code>, and <code class="docutils literal notranslate"><span class="pre">--read_len_2</span></code> controls read length. Specifically:</p>
<ul class="simple">
<li><p>If none of the 3 parameters are set, will use the longest read length supported by the quality distribution file(s).</p></li>
<li><p>If only <code class="docutils literal notranslate"><span class="pre">--read_len</span></code> is set, will use the specified read length for both read 1 and read 2.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--read_len</span></code> cannot be used together with <code class="docutils literal notranslate"><span class="pre">--read_len_1</span></code> or <code class="docutils literal notranslate"><span class="pre">--read_len_2</span></code>.</p></li>
<li><p>If <code class="docutils literal notranslate"><span class="pre">--read_len_1</span></code> and/or <code class="docutils literal notranslate"><span class="pre">--read_len_2</span></code> is set, will use the specified read lengths for read 1 and/or read 2 respectively.</p></li>
<li><p>For paired-end or mate-pair library construction mode, both read lengths must be set either using <code class="docutils literal notranslate"><span class="pre">--read_len</span></code> <strong>OR</strong> <code class="docutils literal notranslate"><span class="pre">--read_len_1</span></code> and <code class="docutils literal notranslate"><span class="pre">--read_len_2</span></code>.</p></li>
</ul>
</section>
<section id="indel-error-parameters">
<h3>Indel Error Parameters<a class="headerlink" href="#indel-error-parameters" title="Link to this heading">¶</a></h3>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">--ins_rate_1</span></code>, <code class="docutils literal notranslate"><span class="pre">--ins_rate_2</span></code>, <code class="docutils literal notranslate"><span class="pre">--del_rate_1</span></code>, and <code class="docutils literal notranslate"><span class="pre">--del_rate_2</span></code>: Targeted insertion and deletion rate for read 1 and 2.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--max_indel</span></code>: The maximum total number of insertions and deletions per read.</p></li>
</ul>
</section>
<section id="ambiguous-base-parameters">
<h3>Ambiguous Base Parameters<a class="headerlink" href="#ambiguous-base-parameters" title="Link to this heading">¶</a></h3>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">--max_n</span></code>: The maximum total number of ambiguous bases (N) per read.</p></li>
</ul>
</section>
<section id="others">
<h3>Others<a class="headerlink" href="#others" title="Link to this heading">¶</a></h3>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">--id</span></code>: Read ID Prefix. Used to distinguish between different runs.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--i-seed</span></code>: Seed for random number generator to ensure reproducibility. Added in <a class="reference internal" href="News.html#v-1-3-2-section"><span class="std std-ref">1.3.2</span></a>. <strong>NOTE</strong> To ensure reproducibility, please use the same number of threads, number of MPI progress under the same random generator when running <code class="docutils literal notranslate"><span class="pre">art_modern</span></code> with the same seed.</p></li>
</ul>
</section>
</section>
<section id="logging-parameters">
<h2>Logging Parameters<a class="headerlink" href="#logging-parameters" title="Link to this heading">¶</a></h2>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">--reporting_interval-job_executor</span></code>: The reporting interval (in seconds) for individual <code class="docutils literal notranslate"><span class="pre">art_modern</span></code> job. Added in <a class="reference internal" href="News.html#v-1-2-1-section"><span class="std std-ref">1.2.1</span></a>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--reporting_interval-job_pool</span></code>: The reporting interval (in seconds) for ThreadPool. Added in <a class="reference internal" href="News.html#v-1-2-1-section"><span class="std std-ref">1.2.1</span></a>.</p></li>
</ul>
<p>Those logs are designed for observation of long-running jobs. Increase the interval to reduce log size.</p>
</section>
<section id="environment-variables">
<span id="am-environment-variables-section"></span><h2>Environment Variables<a class="headerlink" href="#environment-variables" title="Link to this heading">¶</a></h2>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">ART_NO_LOG_DIR</span></code>: If set (to any value), disables creation of a log directory. Logging to files is skipped, and a warning is printed. Added in <a class="reference internal" href="News.html#v-1-1-4-section"><span class="std std-ref">1.1.4</span></a>.</p>
<ul>
<li><p><strong>NOTE</strong> If you’re using <code class="docutils literal notranslate"><span class="pre">art_modern</span></code> with MPI, this discards all logs generated from rank 1, 2, 3….</p></li>
</ul>
</li>
<li><p><code class="docutils literal notranslate"><span class="pre">ART_LOG_DIR</span></code>: If set, specifies the directory where log files will be written. If not set, defaults to <code class="docutils literal notranslate"><span class="pre">log.d</span></code> and a warning is printed. Ignored if <code class="docutils literal notranslate"><span class="pre">ART_NO_LOG_DIR</span></code> is set. Added in <a class="reference internal" href="News.html#v-1-1-4-section"><span class="std std-ref">1.1.4</span></a>.</p>
<ul>
<li><p><strong>NOTE</strong> If you’re running 2 <code class="docutils literal notranslate"><span class="pre">art_modern</span></code> processes simultaneously, please make sure that they are using different log directories.</p></li>
</ul>
</li>
</ul>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="Usage-apb.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">Usage of <code class="docutils literal notranslate"><span class="pre">art_profile_builder</span></code></div>
              </div>
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="Install.html">
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">Installing <code class="docutils literal notranslate"><span class="pre">art_modern</span></code></div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2024-2026, YU Zhejian
            </div>
            Made with <a href="https://www.sphinx-doc.org/">Sphinx</a> and <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
            
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Usage of <code class="docutils literal notranslate"><span class="pre">art_modern</span></code></a><ul>
<li><a class="reference internal" href="#parameter-controlling-simulation-behavior">Parameter Controlling Simulation Behavior</a><ul>
<li><a class="reference internal" href="#simulation-modes-mode">Simulation Modes (<code class="docutils literal notranslate"><span class="pre">--mode</span></code>)</a></li>
<li><a class="reference internal" href="#library-construction-methods-lc">Library Construction Methods (<code class="docutils literal notranslate"><span class="pre">--lc</span></code>)</a><ul>
<li><a class="reference internal" href="#se-default-for-single-end-reads"><strong><code class="docutils literal notranslate"><span class="pre">se</span></code> (DEFAULT) for single-end reads</strong></a></li>
<li><a class="reference internal" href="#pe-for-paired-end-reads"><code class="docutils literal notranslate"><span class="pre">pe</span></code> for paired-end reads</a></li>
<li><a class="reference internal" href="#mp-for-mate-paired-reads"><code class="docutils literal notranslate"><span class="pre">mp</span></code> for mate-paired reads</a></li>
<li><a class="reference internal" href="#see-also">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#thread-level-parallelism-parallel">Thread-Level Parallelism (<code class="docutils literal notranslate"><span class="pre">--parallel</span></code>)</a></li>
<li><a class="reference internal" href="#error-profile-related-parameters">Error-Profile Related Parameters</a><ul>
<li><a class="reference internal" href="#which-quality-distribution-file-to-use">Which Quality Distribution File to Use</a></li>
<li><a class="reference internal" href="#shift-and-clip-parameters">Shift and Clip Parameters</a></li>
<li><a class="reference internal" href="#separate-quality-distributions-for-different-bases-sep-flag">Separate Quality Distributions for Different Bases (<code class="docutils literal notranslate"><span class="pre">--sep_flag</span></code>)</a></li>
</ul>
</li>
</ul>
</li>
<li><a class="reference internal" href="#parameters-controlling-input">Parameters Controlling Input</a><ul>
<li><a class="reference internal" href="#input-file-path-i-file">Input File Path (<code class="docutils literal notranslate"><span class="pre">--i-file</span></code>)</a></li>
<li><a class="reference internal" href="#input-file-type-i-type">Input File Type (<code class="docutils literal notranslate"><span class="pre">--i-type</span></code>)</a></li>
<li><a class="reference internal" href="#input-parser-i-parser">Input Parser (<code class="docutils literal notranslate"><span class="pre">--i-parser</span></code>)</a><ul>
<li><a class="reference internal" href="#auto-default-for-size-based-determination"><strong><code class="docutils literal notranslate"><span class="pre">auto</span></code> (DEFAULT) for size-based determination.</strong></a></li>
<li><a class="reference internal" href="#memory-for-in-memory-reference-parser"><code class="docutils literal notranslate"><span class="pre">memory</span></code> for in-memory reference parser</a></li>
<li><a class="reference internal" href="#htslib-for-htslib-faidx-parser"><code class="docutils literal notranslate"><span class="pre">htslib</span></code> for HTSLib FAIDX parser</a></li>
<li><a class="reference internal" href="#stream-for-one-pass-streaming-parser"><code class="docutils literal notranslate"><span class="pre">stream</span></code> for one-pass streaming parser</a></li>
</ul>
</li>
<li><a class="reference internal" href="#coverage-i-fcov">Coverage (<code class="docutils literal notranslate"><span class="pre">--i-fcov</span></code>)</a></li>
<li><a class="reference internal" href="#compatibility-matrices-of-input-parameters">Compatibility Matrices of Input Parameters</a></li>
</ul>
</li>
<li><a class="reference internal" href="#parameters-controlling-output">Parameters Controlling Output</a><ul>
<li><a class="reference internal" href="#pairwise-alignment-format-o-pwa">Pairwise Alignment Format (<code class="docutils literal notranslate"><span class="pre">--o-pwa</span></code>)</a></li>
<li><a class="reference internal" href="#fastq-format-o-fastq">FASTQ Format (<code class="docutils literal notranslate"><span class="pre">--o-fastq</span></code>)</a></li>
<li><a class="reference internal" href="#fasta-format-o-fasta">FASTA Format (<code class="docutils literal notranslate"><span class="pre">--o-fasta</span></code>)</a></li>
<li><a class="reference internal" href="#sam-bam-format-o-sam">SAM/BAM Format (<code class="docutils literal notranslate"><span class="pre">--o-sam</span></code>)</a></li>
<li><a class="reference internal" href="#headless-sam-bam-format-o-hl-sam">Headless SAM/BAM Format (<code class="docutils literal notranslate"><span class="pre">--o-hl_sam</span></code>)</a></li>
</ul>
</li>
<li><a class="reference internal" href="#miscellaneous-parameters">Miscellaneous Parameters</a><ul>
<li><a class="reference internal" href="#read-length-parameters">Read Length Parameters</a></li>
<li><a class="reference internal" href="#indel-error-parameters">Indel Error Parameters</a></li>
<li><a class="reference internal" href="#ambiguous-base-parameters">Ambiguous Base Parameters</a></li>
<li><a class="reference internal" href="#others">Others</a></li>
</ul>
</li>
<li><a class="reference internal" href="#logging-parameters">Logging Parameters</a></li>
<li><a class="reference internal" href="#environment-variables">Environment Variables</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="../../_static/documentation_options.js?v=e358f374"></script>
    <script src="../../_static/doctools.js?v=fd6eb6e6"></script>
    <script src="../../_static/sphinx_highlight.js?v=6ffebe34"></script>
    <script src="../../_static/scripts/furo.js?v=46bd48cc"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    </body>
</html>